// @vonage/messages/dist/esm/classes/Messenger/Audio.mjs
import {
  MessengerAudio
} from "../../chunk-DLANX2A6.mjs";
import "../../chunk-2VC5OLBL.mjs";
import "../../chunk-YWHURO2W.mjs";
import "../../chunk-SNU746A5.mjs";
import "../../chunk-GTFM4XVI.mjs";
import "../../chunk-4RITRMGD.mjs";
import "../../chunk-57C2DYNH.mjs";
import "../../chunk-SUPGMVFQ.mjs";
import "../../chunk-UIMLGEV7.mjs";
import "../../chunk-CN7QYYRQ.mjs";
import "../../chunk-PGNE2O3U.mjs";
import "../../chunk-QWOWKIZ6.mjs";
import "../../chunk-75B4MJLH.mjs";
import "../../chunk-3WQKOHCE.mjs";

// lib/classes/Messenger/Audio.ts
import debug from "debug";
var log = debug("vonage:messages:messenger");
var Audio = class extends MessengerAudio {
  /**
   * @deprecated Please use MessengerAudio instead.
   * Represents an audio message for the Messenger platform.
   *
   * @param {AudioObject} audio - The audio object containing the audio URL and an optional caption.
   * @param {string} to - The recipient's ID.
   * @param {string} from - The sender's ID.
   * @param {MessageType} messenger - The Messenger message type and category.
   * @param {string} clientRef - An optional client reference.
   */
  constructor(audio, to, from, messenger, clientRef) {
    log("Please update to use the MessengerAudio class instead");
    super({
      audio,
      to,
      from,
      messenger: {
        category: messenger?.category,
        tag: messenger?.tag
      },
      clientRef
    });
  }
};
export {
  Audio
};
