// @vonage/messages/dist/esm/classes/Messenger/Image.mjs
import {
  MessengerImage
} from "../../chunk-6HM2NIVJ.mjs";
import "../../chunk-JXK2KOJV.mjs";
import "../../chunk-YWHURO2W.mjs";
import "../../chunk-SNU746A5.mjs";
import "../../chunk-GTFM4XVI.mjs";
import "../../chunk-4RITRMGD.mjs";
import "../../chunk-57C2DYNH.mjs";
import "../../chunk-SUPGMVFQ.mjs";
import "../../chunk-UIMLGEV7.mjs";
import "../../chunk-CN7QYYRQ.mjs";
import "../../chunk-PGNE2O3U.mjs";
import "../../chunk-QWOWKIZ6.mjs";
import "../../chunk-75B4MJLH.mjs";
import "../../chunk-3WQKOHCE.mjs";

// lib/classes/Messenger/Image.ts
import debug from "debug";
var log = debug("vonage:messages:messenger");
var Image = class extends MessengerImage {
  /**
   * Constructs a new `Image` instance.
   *
   * @param {ImageObject} image - The image object containing the URL and optional caption.
   * @param {string} to - The recipient's ID.
   * @param {string} from - The sender's ID.
   * @param {MessageType} messenger - The messenger object specifying the message category and tag.
   * @param {string} clientRef - The client reference.
   */
  constructor(image, to, from, messenger, clientRef) {
    log("Please update to use the MessengerImage class instead");
    super({
      image,
      to,
      from,
      messenger: {
        category: messenger?.category,
        tag: messenger?.tag
      },
      clientRef
    });
  }
};
export {
  Image
};
