// @vonage/messages/dist/esm/classes/Messenger/Text.mjs
import {
  MessengerText
} from "../../chunk-5XK6RIAI.mjs";
import "../../chunk-LJPOB7FU.mjs";
import "../../chunk-YWHURO2W.mjs";
import "../../chunk-SNU746A5.mjs";
import "../../chunk-GTFM4XVI.mjs";
import "../../chunk-4RITRMGD.mjs";
import "../../chunk-57C2DYNH.mjs";
import "../../chunk-SUPGMVFQ.mjs";
import "../../chunk-UIMLGEV7.mjs";
import "../../chunk-CN7QYYRQ.mjs";
import "../../chunk-PGNE2O3U.mjs";
import "../../chunk-QWOWKIZ6.mjs";
import "../../chunk-75B4MJLH.mjs";
import "../../chunk-3WQKOHCE.mjs";

// lib/classes/Messenger/Text.ts
import debug from "debug";
var log = debug("vonage:messages:messenger");
var Text = class extends MessengerText {
  /**
   * Constructs a new `Text` instance.
   *
   * @param {string} text - The text content of the message.
   * @param {string} to - The recipient of the message.
   * @param {string} from - The sender of the message.
   * @param {MessageType} messenger - The messenger information for the message.
   * @param {string} clientRef - The client reference for the message.
   */
  constructor(text, to, from, messenger, clientRef) {
    log("Please update to use the MessengerText class instead");
    super({
      text,
      to,
      from,
      messenger: {
        category: messenger?.category,
        tag: messenger?.tag
      },
      clientRef
    });
  }
};
export {
  Text
};
