// @vonage/voice/dist/cjs/classes/Endpoint/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/classes/Endpoint/index.ts
var Endpoint_exports = {};
__export(Endpoint_exports, {
  PhoneEndpoint: () => PhoneEndpoint,
  SIPEndpoint: () => SIPEndpoint,
  VBCEndpoint: () => VBCEndpoint,
  WebsocketEndpoint: () => WebsocketEndpoint
});
module.exports = __toCommonJS(Endpoint_exports);

// lib/classes/Endpoint/PhoneEndpoint.ts
var import_debug = __toESM(require("debug"));
(0, import_debug.default)("@vonage/voice")(
  "This class is deprecated. Please update to use the PSTNEndpoint instead"
);
var PhoneEndpoint = class {
  /**
   * The type of the endpoint, which is always 'phone'.
   */
  type;
  /**
   * The phone number associated with this PSTN endpoint.
   */
  number;
  /**
   * Optional DTMF (Dual-Tone Multi-Frequency) answer to send when the call is answered.
   */
  dtmfAnswer;
  /**
   * Create a new PhoneEndpoint instance.
   *
   * @param {string} phoneNumber - The phone number for the PSTN endpoint.
   * @param {string} dtmfAnswer - Optional DTMF answer to send when the call is answered.
   */
  constructor(phoneNumber, dtmfAnswer) {
    this.type = "phone";
    this.number = phoneNumber;
    if (dtmfAnswer) {
      this.dtmfAnswer = dtmfAnswer;
    }
  }
};

// lib/classes/Endpoint/SIPEndpoint.ts
var import_debug2 = __toESM(require("debug"));
(0, import_debug2.default)("@vonage/voice")(
  "This class is deprecated. Please update to use the SIPEndpointType type instead"
);
var SIPEndpoint = class {
  /**
   * The type of the endpoint, which is always 'sip'.
   */
  type;
  /**
   * The SIP URI associated with this endpoint.
   *
   * @param {string} uri - The SIP URI for the SIP endpoint.
   */
  uri;
  /**
   * Optional custom headers to include in SIP requests.
   *
   * @param {Array<Record<string, unknown>>} headers - Optional custom headers as an array of key-value pairs.
   */
  headers;
  /**
   * Create a new SIPEndpoint instance.
   *
   * @param {string} uri - The SIP URI for the SIP endpoint.
   * @param {Array<Record<string, unknown>>} headers - Optional custom headers as an array of key-value pairs.
   */
  constructor(uri, headers) {
    this.type = "sip";
    this.uri = uri;
    if (headers) {
      this.headers = headers[0];
    }
  }
};

// lib/classes/Endpoint/VBCEndpoint.ts
var import_debug3 = __toESM(require("debug"));
(0, import_debug3.default)("@vonage/voice")(
  "This class is deprecated. Please update to use the VBCEndpointType type instead"
);
var VBCEndpoint = class {
  /**
   * The type of the endpoint, which is always 'vbc'.
   */
  type;
  /**
   * The VBC extension associated with this endpoint.
   *
   * @param {string} extension - The VBC extension for the VBC endpoint.
   */
  extension;
  /**
   * Create a new VBCEndpoint instance.
   *
   * @param {string} extension - The VBC extension for the VBC endpoint.
   */
  constructor(extension) {
    this.type = "vbc";
    this.extension = extension;
  }
};

// lib/classes/Endpoint/WebsocketEndpoint.ts
var import_debug4 = __toESM(require("debug"));
(0, import_debug4.default)("@vonage/voice")(
  "This class is deprecated. Please update to use the WebsocketEndpointType type instead"
);
var WebsocketEndpoint = class {
  /**
   * The type of the endpoint, which is always 'websocket'.
   */
  type;
  /**
   * The WebSocket URI associated with this endpoint.
   *
   * @param {string} uri - The WebSocket URI for the WebSocket endpoint.
   */
  uri;
  /**
   * The content type or bitrate for WebSocket streaming.
   *
   * @param {WebsocketBitrate} contentType - The content type or bitrate for WebSocket streaming.
   */
  contentType;
  /**
   * Optional custom headers to include in WebSocket requests.
   *
   * @param {Record<string, unknown>} headers - Optional custom headers as key-value pairs.
   */
  headers;
  /**
   * Create a new WebsocketEndpoint instance.
   *
   * @param {string} uri - The WebSocket URI for the WebSocket endpoint.
   * @param {WebsocketBitrate} contentType - The content type or bitrate for WebSocket streaming.
   * @param {Record<string, unknown>} headers - Optional custom headers as key-value pairs.
   */
  constructor(uri, contentType, headers) {
    this.type = "websocket";
    this.uri = uri;
    this.contentType = contentType;
    if (headers) {
      this.headers = headers;
    }
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PhoneEndpoint,
  SIPEndpoint,
  VBCEndpoint,
  WebsocketEndpoint
});
