// @vonage/messages/dist/cjs/enums/Messenger/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/Messenger/index.ts
var Messenger_exports = {};
__export(Messenger_exports, {
  MessengerCategory: () => MessengerCategory,
  MessengerTags: () => MessengerTags
});
module.exports = __toCommonJS(Messenger_exports);

// lib/enums/Messenger/MessengerCategory.ts
var MessengerCategory = /* @__PURE__ */ ((MessengerCategory2) => {
  MessengerCategory2["RESPONSE"] = "response";
  MessengerCategory2["UPDATE"] = "update";
  MessengerCategory2["MESSAGE_TAG"] = "message_tag";
  return MessengerCategory2;
})(MessengerCategory || {});

// lib/enums/Messenger/MessengerTags.ts
var MessengerTags = /* @__PURE__ */ ((MessengerTags2) => {
  MessengerTags2["CONFIRMED_EVENT_UPDATE"] = "CONFIRMED_EVENT_UPDATE";
  MessengerTags2["POST_PURCHASE_UPDATE"] = "POST_PURCHASE_UPDATE";
  MessengerTags2["ACCOUNT_UPDATE"] = "ACCOUNT_UPDATE";
  MessengerTags2["HUMAN_AGENT"] = "HUMAN_AGENT";
  MessengerTags2["SHIPPING_UPDATE"] = "SHIPPING_UPDATE";
  MessengerTags2["RESERVATION_UPDATE"] = "RESERVATION_UPDATE";
  MessengerTags2["ISSUE_RESOLUTION"] = "ISSUE_RESOLUTION";
  MessengerTags2["APPOINTMENT_UPDATE"] = "APPOINTMENT_UPDATE";
  MessengerTags2["GAME_EVENT"] = "GAME_EVENT";
  MessengerTags2["TRANSPORTATION_UPDATE"] = "TRANSPORTATION_UPDATE";
  MessengerTags2["FEATURE_FUNCTIONALITY_UPDATE"] = "FEATURE_FUNCTIONALITY_UPDATE";
  MessengerTags2["TICKET_UPDATE"] = "TICKET_UPDATE";
  MessengerTags2["PAYMENT_UPDATE"] = "PAYMENT_UPDATE";
  MessengerTags2["PERSONAL_FINANCE_UPDATE"] = "PERSONAL_FINANCE_UPDATE";
  return MessengerTags2;
})(MessengerTags || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MessengerCategory,
  MessengerTags
});
