// @vonage/voice/dist/esm/chunk-Y3T6C6QB.mjs
// lib/enums/TTSLanguages.ts
var TTSLanguages = /* @__PURE__ */ ((TTSLanguages2) => {
  TTSLanguages2["AF_ZA"] = "af-ZA";
  TTSLanguages2["AR"] = "ar";
  TTSLanguages2["BG_BG"] = "bg-BG";
  TTSLanguages2["BN_IN"] = "bn-IN";
  TTSLanguages2["CA_ES"] = "ca-ES";
  TTSLanguages2["CMN_CN"] = "cmn-CN";
  TTSLanguages2["CMN_TW"] = "cmn-TW";
  TTSLanguages2["CS_CZ"] = "cs-CZ";
  TTSLanguages2["CY_GB"] = "cy-GB";
  TTSLanguages2["DA_DK"] = "da-DK";
  TTSLanguages2["DE_AT"] = "de-AT";
  TTSLanguages2["DE_DE"] = "de-DE";
  TTSLanguages2["DE_ET"] = "de-ET";
  TTSLanguages2["EL_GR"] = "el-GR";
  TTSLanguages2["EN_AU"] = "en-AU";
  TTSLanguages2["EN_GB"] = "en-GB";
  TTSLanguages2["EN_GB_SCT"] = "en-GB-SCT";
  TTSLanguages2["EN_GB_WLS"] = "en-GB-WLS";
  TTSLanguages2["EN_IE"] = "en-IE";
  TTSLanguages2["EN_IN"] = "en-IN";
  TTSLanguages2["EN_NZ"] = "en-NZ";
  TTSLanguages2["EN_US"] = "en-US";
  TTSLanguages2["EN_ZA"] = "en-ZA";
  TTSLanguages2["ES_CO"] = "es-CO";
  TTSLanguages2["ES_ES"] = "es-ES";
  TTSLanguages2["ES_MX"] = "es-MX";
  TTSLanguages2["ES_US"] = "es-US";
  TTSLanguages2["EU_ES"] = "eu-ES";
  TTSLanguages2["FI_FI"] = "fi-FI";
  TTSLanguages2["FIL_PH"] = "fil-PH";
  TTSLanguages2["FR_CA"] = "fr-CA";
  TTSLanguages2["FR_FR"] = "fr-FR";
  TTSLanguages2["GL_ES"] = "gl-ES";
  TTSLanguages2["GU_IN"] = "gu-IN";
  TTSLanguages2["HE_IL"] = "he-IL";
  TTSLanguages2["HI_IN"] = "hi-IN";
  TTSLanguages2["HU_HU"] = "hu-HU";
  TTSLanguages2["ID_ID"] = "id-ID";
  TTSLanguages2["IS_IS"] = "is-IS";
  TTSLanguages2["IT_IT"] = "it-IT";
  TTSLanguages2["JA_JP"] = "ja-JP";
  TTSLanguages2["KN_IN"] = "kn-IN";
  TTSLanguages2["KO_KR"] = "ko-KR";
  TTSLanguages2["LV_LV"] = "lv-LV";
  TTSLanguages2["ML_IN"] = "ml-IN";
  TTSLanguages2["MS_MY"] = "ms-MY";
  TTSLanguages2["NB_NO"] = "nb-NO";
  TTSLanguages2["NL_BE"] = "nl-BE";
  TTSLanguages2["NL_NL"] = "nl-NL";
  TTSLanguages2["NO_NO"] = "no-NO";
  TTSLanguages2["PA_IN"] = "pa-IN";
  TTSLanguages2["PL_PL"] = "pl-PL";
  TTSLanguages2["PT_BR"] = "pt-BR";
  TTSLanguages2["PT_PT"] = "pt-PT";
  TTSLanguages2["RO_RO"] = "ro-RO";
  TTSLanguages2["RU_RU"] = "ru-RU";
  TTSLanguages2["SK_SK"] = "sk-SK";
  TTSLanguages2["SR_RS"] = "sr-RS";
  TTSLanguages2["SV_SE"] = "sv-SE";
  TTSLanguages2["TA_IN"] = "ta-IN";
  TTSLanguages2["TE_IN"] = "te-IN";
  TTSLanguages2["TH_TH"] = "th-TH";
  TTSLanguages2["TR_TR"] = "tr-TR";
  TTSLanguages2["UK_UA"] = "uk-UA";
  TTSLanguages2["VI_VN"] = "vi-VN";
  TTSLanguages2["YUE_CN"] = "yue-CN";
  return TTSLanguages2;
})(TTSLanguages || {});

export {
  TTSLanguages
};
