// @vonage/voice/dist/cjs/enums/NCCO/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/NCCO/index.ts
var NCCO_exports = {};
__export(NCCO_exports, {
  ConnectEventType: () => ConnectEventType,
  MachineDetection: () => MachineDetection,
  RecordingFormat: () => RecordingFormat
});
module.exports = __toCommonJS(NCCO_exports);

// lib/enums/NCCO/ConnectEventType.ts
var ConnectEventType = /* @__PURE__ */ ((ConnectEventType2) => {
  ConnectEventType2["SYNCHRONOUS"] = "synchronous";
  return ConnectEventType2;
})(ConnectEventType || {});

// lib/enums/NCCO/MachineDetection.ts
var MachineDetection = /* @__PURE__ */ ((MachineDetection2) => {
  MachineDetection2["CONTINUE"] = "continue";
  MachineDetection2["HANGUP"] = "hangup";
  return MachineDetection2;
})(MachineDetection || {});

// lib/enums/NCCO/RecordingFormat.ts
var RecordingFormat = /* @__PURE__ */ ((RecordingFormat2) => {
  RecordingFormat2["MP3"] = "mp3";
  RecordingFormat2["WAV"] = "wav";
  RecordingFormat2["OGG"] = "ogg";
  return RecordingFormat2;
})(RecordingFormat || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ConnectEventType,
  MachineDetection,
  RecordingFormat
});
