// @vonage/verify/dist/cjs/enums/CheckStatus.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/CheckStatus.ts
var CheckStatus_exports = {};
__export(CheckStatus_exports, {
  CheckStatus: () => CheckStatus
});
module.exports = __toCommonJS(CheckStatus_exports);
var CheckStatus = /* @__PURE__ */ ((CheckStatus2) => {
  CheckStatus2["SUCCESS"] = "0";
  CheckStatus2["THROTTLED"] = "1";
  CheckStatus2["MISSING_REQUIRED_PARAM"] = "2";
  CheckStatus2["INVALID_PARAM"] = "3";
  CheckStatus2["INVALID_CREDENTIALS"] = "4";
  CheckStatus2["INTERNAL_ERROR"] = "5";
  CheckStatus2["FAILED_TO_PROCESS"] = "6";
  CheckStatus2["BARRED_API_KEY"] = "8";
  CheckStatus2["PARTNER_QUOTA_EXCEEDED"] = "9";
  CheckStatus2["CONCURRENT_VERIFICATIONS"] = "10";
  CheckStatus2["UNSUPPORTED_NETWORK"] = "15";
  CheckStatus2["CODE_MISMATCH"] = "16";
  CheckStatus2["INVALID_CODE"] = "17";
  CheckStatus2["DELIVERY_FAILED"] = "19";
  CheckStatus2["PIN_NOT_SUPPORTED"] = "20";
  CheckStatus2["NON_PERMITTED_DESTINATION"] = "29";
  return CheckStatus2;
})(CheckStatus || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CheckStatus
});
