// @vonage/verify/dist/cjs/enums/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/index.ts
var enums_exports = {};
__export(enums_exports, {
  CheckStatus: () => CheckStatus,
  Command: () => Command,
  SearchCheckStatus: () => SearchCheckStatus,
  SearchEventTypes: () => SearchEventTypes,
  SearchStatus: () => SearchStatus,
  VerifyLanguages: () => VerifyLanguages,
  VerifyWorkflows: () => VerifyWorkflows
});
module.exports = __toCommonJS(enums_exports);

// lib/enums/CheckStatus.ts
var CheckStatus = /* @__PURE__ */ ((CheckStatus2) => {
  CheckStatus2["SUCCESS"] = "0";
  CheckStatus2["THROTTLED"] = "1";
  CheckStatus2["MISSING_REQUIRED_PARAM"] = "2";
  CheckStatus2["INVALID_PARAM"] = "3";
  CheckStatus2["INVALID_CREDENTIALS"] = "4";
  CheckStatus2["INTERNAL_ERROR"] = "5";
  CheckStatus2["FAILED_TO_PROCESS"] = "6";
  CheckStatus2["BARRED_API_KEY"] = "8";
  CheckStatus2["PARTNER_QUOTA_EXCEEDED"] = "9";
  CheckStatus2["CONCURRENT_VERIFICATIONS"] = "10";
  CheckStatus2["UNSUPPORTED_NETWORK"] = "15";
  CheckStatus2["CODE_MISMATCH"] = "16";
  CheckStatus2["INVALID_CODE"] = "17";
  CheckStatus2["DELIVERY_FAILED"] = "19";
  CheckStatus2["PIN_NOT_SUPPORTED"] = "20";
  CheckStatus2["NON_PERMITTED_DESTINATION"] = "29";
  return CheckStatus2;
})(CheckStatus || {});

// lib/enums/Command.ts
var Command = /* @__PURE__ */ ((Command2) => {
  Command2["CANCEL"] = "cancel";
  Command2["TRIGGER_NEXT_EVENT"] = "trigger_next_event";
  return Command2;
})(Command || {});

// lib/enums/SearchCheckStatus.ts
var SearchCheckStatus = /* @__PURE__ */ ((SearchCheckStatus2) => {
  SearchCheckStatus2["INVALID"] = "INVALID";
  SearchCheckStatus2["VALID"] = "VALID";
  return SearchCheckStatus2;
})(SearchCheckStatus || {});

// lib/enums/SearchEventTypes.ts
var SearchEventTypes = /* @__PURE__ */ ((SearchEventTypes2) => {
  SearchEventTypes2["TTS"] = "TTS";
  SearchEventTypes2["SMS"] = "SMS";
  return SearchEventTypes2;
})(SearchEventTypes || {});

// lib/enums/SearchStatus.ts
var SearchStatus = /* @__PURE__ */ ((SearchStatus2) => {
  SearchStatus2["IN_PROGRESS"] = "IN PROGRESS";
  SearchStatus2["SUCCESS"] = "SUCCESS";
  SearchStatus2["FAILED"] = "FAILED";
  SearchStatus2["EXPIRED"] = "EXPIRED";
  SearchStatus2["CANCELLED"] = "CANCELLED";
  return SearchStatus2;
})(SearchStatus || {});

// lib/enums/VerifyLanguages.ts
var VerifyLanguages = /* @__PURE__ */ ((VerifyLanguages2) => {
  VerifyLanguages2["ARABIC"] = "ar-xa";
  VerifyLanguages2["CZECH"] = "cs-cz";
  VerifyLanguages2["WELSH"] = "cy-cy";
  VerifyLanguages2["WELSH_UK"] = "cy-gb";
  VerifyLanguages2["DANISH"] = "da-dk";
  VerifyLanguages2["GERMAN"] = "de-de";
  VerifyLanguages2["GREEK"] = "el-gr";
  VerifyLanguages2["ENGLISH_AUSTRALIA"] = "en-au";
  VerifyLanguages2["ENGLISH_UK"] = "en-gb";
  VerifyLanguages2["ENGLISH_INDIA"] = "en-in";
  VerifyLanguages2["ENGLISH_US"] = "en-us";
  VerifyLanguages2["SPANISH_SPAIN"] = "es-es";
  VerifyLanguages2["SPANISH_MEXICO"] = "es-mx";
  VerifyLanguages2["SPANISH_US"] = "es-us";
  VerifyLanguages2["FINNISH"] = "fi-fi";
  VerifyLanguages2["FILIPINO"] = "fil-ph";
  VerifyLanguages2["FRENCH_CANADA"] = "fr-ca";
  VerifyLanguages2["FRENCH_FRANCE"] = "fr-fr";
  VerifyLanguages2["HINDI"] = "hi-in";
  VerifyLanguages2["HUNGARIAN"] = "hu-hu";
  VerifyLanguages2["INDONESIAN"] = "id-id";
  VerifyLanguages2["ICELANDIC"] = "is-is";
  VerifyLanguages2["ITALIAN"] = "it-it";
  VerifyLanguages2["JAPANESE"] = "ja-jp";
  VerifyLanguages2["KOREAN"] = "ko-kr";
  VerifyLanguages2["NORWEGIAN"] = "nb-no";
  VerifyLanguages2["DUTCH"] = "nl-nl";
  VerifyLanguages2["POLISH"] = "pl-pl";
  VerifyLanguages2["PORTUGUESE_BRAZIL"] = "pt-br";
  VerifyLanguages2["PORTUGUESE_PORTUGAL"] = "pt-pt";
  VerifyLanguages2["ROMANIAN"] = "ro-ro";
  VerifyLanguages2["RUSSIAN"] = "ru-ru";
  VerifyLanguages2["SWEDISH"] = "sv-se";
  VerifyLanguages2["THAI"] = "th-th";
  VerifyLanguages2["TURKISH"] = "tr-tr";
  VerifyLanguages2["VIETNAMESE"] = "vi-vn";
  VerifyLanguages2["YUE_CHINESE_CANTONESE"] = "yue-cn";
  VerifyLanguages2["CHINESE_MAINLAND"] = "zh-cn";
  VerifyLanguages2["CHINESE_TAIWAN"] = "zh-tw";
  return VerifyLanguages2;
})(VerifyLanguages || {});

// lib/enums/Workflows.ts
var VerifyWorkflows = /* @__PURE__ */ ((VerifyWorkflows2) => {
  VerifyWorkflows2["SMS_TTS_TTS"] = "1";
  VerifyWorkflows2["SMS_SMS_TTS"] = "2";
  VerifyWorkflows2["TTS_TTS"] = "3";
  VerifyWorkflows2["SMS_SMS"] = "4";
  VerifyWorkflows2["SMS_TTS"] = "5";
  VerifyWorkflows2["SMS"] = "6";
  VerifyWorkflows2["TTS"] = "7";
  return VerifyWorkflows2;
})(VerifyWorkflows || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CheckStatus,
  Command,
  SearchCheckStatus,
  SearchEventTypes,
  SearchStatus,
  VerifyLanguages,
  VerifyWorkflows
});
