// @vonage/verify/dist/cjs/enums/VerifyStatus.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/VerifyStatus.ts
var VerifyStatus_exports = {};
__export(VerifyStatus_exports, {
  VerifyStatus: () => VerifyStatus
});
module.exports = __toCommonJS(VerifyStatus_exports);
var VerifyStatus = /* @__PURE__ */ ((VerifyStatus2) => {
  VerifyStatus2["SUCCESS"] = "0";
  VerifyStatus2["THROTTLED"] = "1";
  VerifyStatus2["MISSING_REQUIRED_PARAM"] = "2";
  VerifyStatus2["INVALID_PARAM"] = "3";
  VerifyStatus2["INVALID_CREDENTIALS"] = "4";
  VerifyStatus2["INTERNAL_ERROR"] = "5";
  VerifyStatus2["FAILED_TO_PROCESS"] = "6";
  VerifyStatus2["BARRED_API_KEY"] = "8";
  VerifyStatus2["PARTNER_QUOTA_EXCEEDED"] = "9";
  VerifyStatus2["CONCURRENT_VERIFICATIONS"] = "10";
  VerifyStatus2["UNSUPPORTED_NETWORK"] = "15";
  VerifyStatus2["CODE_MISMATCH"] = "16";
  VerifyStatus2["INVALID_CODE"] = "17";
  VerifyStatus2["DELIVERY_FAILED"] = "19";
  VerifyStatus2["PIN_NOT_SUPPORTED"] = "20";
  VerifyStatus2["NON_PERMITTED_DESTINATION"] = "29";
  return VerifyStatus2;
})(VerifyStatus || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  VerifyStatus
});
