// @vonage/verify/dist/cjs/enums/Workflows.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/Workflows.ts
var Workflows_exports = {};
__export(Workflows_exports, {
  VerifyWorkflows: () => VerifyWorkflows
});
module.exports = __toCommonJS(Workflows_exports);
var VerifyWorkflows = /* @__PURE__ */ ((VerifyWorkflows2) => {
  VerifyWorkflows2["SMS_TTS_TTS"] = "1";
  VerifyWorkflows2["SMS_SMS_TTS"] = "2";
  VerifyWorkflows2["TTS_TTS"] = "3";
  VerifyWorkflows2["SMS_SMS"] = "4";
  VerifyWorkflows2["SMS_TTS"] = "5";
  VerifyWorkflows2["SMS"] = "6";
  VerifyWorkflows2["TTS"] = "7";
  return VerifyWorkflows2;
})(VerifyWorkflows || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  VerifyWorkflows
});
