// @vonage/sms/dist/esm/chunk-PVKVH2ES.mjs
// lib/enums/SMSStatus.ts
var SMSStatus = /* @__PURE__ */ ((SMSStatus2) => {
  SMSStatus2["SUCCESS"] = "0";
  SMSStatus2["THROTTLED"] = "1";
  SMSStatus2["MISSING_REQUIRED_PARAM"] = "2";
  SMSStatus2["INVALID_PARAM"] = "3";
  SMSStatus2["INVALID_CREDENTIALS"] = "4";
  SMSStatus2["INTERNAL_ERROR"] = "5";
  SMSStatus2["INVALID_MESSAGE"] = "6";
  SMSStatus2["BARRED_NUMBER"] = "7";
  SMSStatus2["PARTNER_ACCOUNT_BARRED"] = "8";
  SMSStatus2["PARTNER_QUOTA_VIOLATION"] = "9";
  SMSStatus2["TOO_MANY_EXISTING_BINDS"] = "10";
  SMSStatus2["ACCOUNT_NOT_ENABLED"] = "11";
  SMSStatus2["MESSAGE_TOO_LONG"] = "12";
  SMSStatus2["INVALID_SIGNATURE"] = "14";
  SMSStatus2["INVALID_SENDER_ADDRESS"] = "15";
  SMSStatus2["INVALID_NETWORK_CODE"] = "22";
  SMSStatus2["INVALID_CALLBACK_URL"] = "23";
  SMSStatus2["NON_WHITELISTED_DESTINATION"] = "29";
  SMSStatus2["SIGNATURE_AND_API_SECRET_DISALLOWED"] = "32";
  SMSStatus2["NUMBER_DEACTIVATED"] = "33";
  return SMSStatus2;
})(SMSStatus || {});

export {
  SMSStatus
};
