// @vonage/messages/dist/cjs/enums/Channels.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/Channels.ts
var Channels_exports = {};
__export(Channels_exports, {
  Channels: () => Channels
});
module.exports = __toCommonJS(Channels_exports);
var Channels = /* @__PURE__ */ ((Channels2) => {
  Channels2["MESSENGER"] = "messenger";
  Channels2["MMS"] = "mms";
  Channels2["SMS"] = "sms";
  Channels2["WHATSAPP"] = "whatsapp";
  Channels2["VIBER"] = "viber_service";
  Channels2["RCS"] = "rcs";
  return Channels2;
})(Channels || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Channels
});
