// @vonage/messages/dist/cjs/enums/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/index.ts
var enums_exports = {};
__export(enums_exports, {
  Channels: () => Channels,
  MessageTypes: () => MessageTypes,
  MessengerCategory: () => MessengerCategory,
  MessengerTags: () => MessengerTags,
  UpdateMessageStatus: () => UpdateMessageStatus,
  ViberCategory: () => ViberCategory,
  WhatsAppLanguageCode: () => WhatsAppLanguageCode
});
module.exports = __toCommonJS(enums_exports);

// lib/enums/Channels.ts
var Channels = /* @__PURE__ */ ((Channels2) => {
  Channels2["MESSENGER"] = "messenger";
  Channels2["MMS"] = "mms";
  Channels2["SMS"] = "sms";
  Channels2["WHATSAPP"] = "whatsapp";
  Channels2["VIBER"] = "viber_service";
  Channels2["RCS"] = "rcs";
  return Channels2;
})(Channels || {});

// lib/enums/MessageTypes.ts
var MessageTypes = /* @__PURE__ */ ((MessageTypes2) => {
  MessageTypes2["TEXT"] = "text";
  MessageTypes2["IMAGE"] = "image";
  MessageTypes2["VCARD"] = "vcard";
  MessageTypes2["AUDIO"] = "audio";
  MessageTypes2["VIDEO"] = "video";
  MessageTypes2["FILE"] = "file";
  MessageTypes2["CONTENT"] = "content";
  MessageTypes2["CUSTOM"] = "custom";
  MessageTypes2["TEMPLATE"] = "template";
  MessageTypes2["STICKER"] = "sticker";
  MessageTypes2["REACTION"] = "reaction";
  return MessageTypes2;
})(MessageTypes || {});

// lib/enums/Messenger/MessengerCategory.ts
var MessengerCategory = /* @__PURE__ */ ((MessengerCategory2) => {
  MessengerCategory2["RESPONSE"] = "response";
  MessengerCategory2["UPDATE"] = "update";
  MessengerCategory2["MESSAGE_TAG"] = "message_tag";
  return MessengerCategory2;
})(MessengerCategory || {});

// lib/enums/Messenger/MessengerTags.ts
var MessengerTags = /* @__PURE__ */ ((MessengerTags2) => {
  MessengerTags2["CONFIRMED_EVENT_UPDATE"] = "CONFIRMED_EVENT_UPDATE";
  MessengerTags2["POST_PURCHASE_UPDATE"] = "POST_PURCHASE_UPDATE";
  MessengerTags2["ACCOUNT_UPDATE"] = "ACCOUNT_UPDATE";
  MessengerTags2["HUMAN_AGENT"] = "HUMAN_AGENT";
  MessengerTags2["SHIPPING_UPDATE"] = "SHIPPING_UPDATE";
  MessengerTags2["RESERVATION_UPDATE"] = "RESERVATION_UPDATE";
  MessengerTags2["ISSUE_RESOLUTION"] = "ISSUE_RESOLUTION";
  MessengerTags2["APPOINTMENT_UPDATE"] = "APPOINTMENT_UPDATE";
  MessengerTags2["GAME_EVENT"] = "GAME_EVENT";
  MessengerTags2["TRANSPORTATION_UPDATE"] = "TRANSPORTATION_UPDATE";
  MessengerTags2["FEATURE_FUNCTIONALITY_UPDATE"] = "FEATURE_FUNCTIONALITY_UPDATE";
  MessengerTags2["TICKET_UPDATE"] = "TICKET_UPDATE";
  MessengerTags2["PAYMENT_UPDATE"] = "PAYMENT_UPDATE";
  MessengerTags2["PERSONAL_FINANCE_UPDATE"] = "PERSONAL_FINANCE_UPDATE";
  return MessengerTags2;
})(MessengerTags || {});

// lib/enums/UpdateMessageStatus.ts
var UpdateMessageStatus = /* @__PURE__ */ ((UpdateMessageStatus2) => {
  UpdateMessageStatus2["READ"] = "read";
  UpdateMessageStatus2["REVOKED"] = "revoked";
  return UpdateMessageStatus2;
})(UpdateMessageStatus || {});

// lib/enums/Viber/ViberCategory.ts
var ViberCategory = /* @__PURE__ */ ((ViberCategory2) => {
  ViberCategory2["TRANSACTION"] = "transaction";
  ViberCategory2["PROMOTION"] = "promotion";
  return ViberCategory2;
})(ViberCategory || {});

// lib/enums/WhatsApp/WhatsAppLanguageCodes.ts
var WhatsAppLanguageCode = /* @__PURE__ */ ((WhatsAppLanguageCode2) => {
  WhatsAppLanguageCode2["AFRIKAANS"] = "af";
  WhatsAppLanguageCode2["ALBANIAN"] = "sq";
  WhatsAppLanguageCode2["ARABIC"] = "ar";
  WhatsAppLanguageCode2["AZERBAIJANI"] = "az";
  WhatsAppLanguageCode2["BENGALI"] = "bn";
  WhatsAppLanguageCode2["BULGARIAN"] = "bg";
  WhatsAppLanguageCode2["CATALAN"] = "ca";
  WhatsAppLanguageCode2["CHINESE_CHN"] = "zh_CN";
  WhatsAppLanguageCode2["CHINESE_HKG"] = "zh_HK";
  WhatsAppLanguageCode2["CHINESE_TAI"] = "zh_TW";
  WhatsAppLanguageCode2["CROATIAN"] = "hr";
  WhatsAppLanguageCode2["CZECH"] = "cs";
  WhatsAppLanguageCode2["DANISH"] = "da";
  WhatsAppLanguageCode2["DUTCH"] = "nl";
  WhatsAppLanguageCode2["ENGLISH"] = "en";
  WhatsAppLanguageCode2["ENGLISH_UK"] = "en_GB";
  WhatsAppLanguageCode2["ENGLISH_US"] = "en_US";
  WhatsAppLanguageCode2["ESTONIAN"] = "et";
  WhatsAppLanguageCode2["FILIPINO"] = "fil";
  WhatsAppLanguageCode2["FINNISH"] = "fi";
  WhatsAppLanguageCode2["FRENCH"] = "fr";
  WhatsAppLanguageCode2["GEORGIAN"] = "ka";
  WhatsAppLanguageCode2["GERMAN"] = "de";
  WhatsAppLanguageCode2["GREEK"] = "el";
  WhatsAppLanguageCode2["GUJARATI"] = "gu";
  WhatsAppLanguageCode2["HAUSA"] = "ha";
  WhatsAppLanguageCode2["HEBREW"] = "he";
  WhatsAppLanguageCode2["HINDI"] = "hi";
  WhatsAppLanguageCode2["HUNGARIAN"] = "hu";
  WhatsAppLanguageCode2["INDONESIAN"] = "id";
  WhatsAppLanguageCode2["IRISH"] = "ga";
  WhatsAppLanguageCode2["ITALIAN"] = "it";
  WhatsAppLanguageCode2["JAPANESE"] = "ja";
  WhatsAppLanguageCode2["KANNADA"] = "kn";
  WhatsAppLanguageCode2["KAZAKH"] = "kk";
  WhatsAppLanguageCode2["KINYARWANDA"] = "rw_RW";
  WhatsAppLanguageCode2["KOREAN"] = "ko";
  WhatsAppLanguageCode2["KYRGYZ_KYRGYZSTAN"] = "ky_KG";
  WhatsAppLanguageCode2["LAO"] = "lo";
  WhatsAppLanguageCode2["LATVIAN"] = "lv";
  WhatsAppLanguageCode2["LITHUANIAN"] = "lt";
  WhatsAppLanguageCode2["MACEDONIAN"] = "mk";
  WhatsAppLanguageCode2["MALAY"] = "ms";
  WhatsAppLanguageCode2["MALAYALAM"] = "ml";
  WhatsAppLanguageCode2["MARATHI"] = "mr";
  WhatsAppLanguageCode2["NORWEGIAN"] = "nb";
  WhatsAppLanguageCode2["PERSIAN"] = "fa";
  WhatsAppLanguageCode2["POLISH"] = "pl";
  WhatsAppLanguageCode2["PORTUGUESE_BR"] = "pt_BR";
  WhatsAppLanguageCode2["PORTUGUESE_POR"] = "pt_PT";
  WhatsAppLanguageCode2["PUNJABI"] = "pa";
  WhatsAppLanguageCode2["ROMANIAN"] = "ro";
  WhatsAppLanguageCode2["RUSSIAN"] = "ru";
  WhatsAppLanguageCode2["SERBIAN"] = "sr";
  WhatsAppLanguageCode2["SLOVAK"] = "sk";
  WhatsAppLanguageCode2["SLOVENIAN"] = "sl";
  WhatsAppLanguageCode2["SPANISH"] = "es";
  WhatsAppLanguageCode2["SPANISH_ARG"] = "es_AR";
  WhatsAppLanguageCode2["SPANISH_SPA"] = "es_ES";
  WhatsAppLanguageCode2["SPANISH_MEX"] = "es_MX";
  WhatsAppLanguageCode2["SWAHILI"] = "sw";
  WhatsAppLanguageCode2["SWEDISH"] = "sv";
  WhatsAppLanguageCode2["TAMIL"] = "ta";
  WhatsAppLanguageCode2["TELUGU"] = "te";
  WhatsAppLanguageCode2["THAI"] = "th";
  WhatsAppLanguageCode2["TURKISH"] = "tr";
  WhatsAppLanguageCode2["UKRAINIAN"] = "uk";
  WhatsAppLanguageCode2["URDU"] = "ur";
  WhatsAppLanguageCode2["UZBEK"] = "uz";
  WhatsAppLanguageCode2["VIETNAMESE"] = "vi";
  WhatsAppLanguageCode2["ZULU"] = "zu";
  return WhatsAppLanguageCode2;
})(WhatsAppLanguageCode || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Channels,
  MessageTypes,
  MessengerCategory,
  MessengerTags,
  UpdateMessageStatus,
  ViberCategory,
  WhatsAppLanguageCode
});
