// @vonage/messages/dist/cjs/enums/MessageTypes.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/MessageTypes.ts
var MessageTypes_exports = {};
__export(MessageTypes_exports, {
  MessageTypes: () => MessageTypes
});
module.exports = __toCommonJS(MessageTypes_exports);
var MessageTypes = /* @__PURE__ */ ((MessageTypes2) => {
  MessageTypes2["TEXT"] = "text";
  MessageTypes2["IMAGE"] = "image";
  MessageTypes2["VCARD"] = "vcard";
  MessageTypes2["AUDIO"] = "audio";
  MessageTypes2["VIDEO"] = "video";
  MessageTypes2["FILE"] = "file";
  MessageTypes2["CONTENT"] = "content";
  MessageTypes2["CUSTOM"] = "custom";
  MessageTypes2["TEMPLATE"] = "template";
  MessageTypes2["STICKER"] = "sticker";
  MessageTypes2["REACTION"] = "reaction";
  return MessageTypes2;
})(MessageTypes || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MessageTypes
});
