// @vonage/voice/dist/cjs/enums/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/index.ts
var enums_exports = {};
__export(enums_exports, {
  AdvancedMachineDetectionMode: () => AdvancedMachineDetectionMode,
  CallDirection: () => CallDirection,
  CallStatus: () => CallStatus,
  ConnectEventType: () => ConnectEventType,
  HttpMethod: () => HttpMethod,
  MachineDetection: () => MachineDetection,
  MachineDetectionBehavior: () => MachineDetectionBehavior,
  NCCOActions: () => NCCOActions,
  RecordingFormat: () => RecordingFormat,
  TTSLanguages: () => TTSLanguages,
  TTSVoices: () => TTSVoices,
  WebsocketBitrate: () => WebsocketBitrate
});
module.exports = __toCommonJS(enums_exports);

// lib/enums/AdvancedMachineDetectionMode.ts
var AdvancedMachineDetectionMode = /* @__PURE__ */ ((AdvancedMachineDetectionMode2) => {
  AdvancedMachineDetectionMode2["DEFAULT"] = "default";
  AdvancedMachineDetectionMode2["DETECT"] = "detect";
  AdvancedMachineDetectionMode2["DETECTBEEP"] = "detect_beep";
  return AdvancedMachineDetectionMode2;
})(AdvancedMachineDetectionMode || {});

// lib/enums/CallDirection.ts
var CallDirection = /* @__PURE__ */ ((CallDirection2) => {
  CallDirection2["OUTBOUND"] = "outbound";
  CallDirection2["INBOUND"] = "inbound";
  return CallDirection2;
})(CallDirection || {});

// lib/enums/CallStatus.ts
var CallStatus = /* @__PURE__ */ ((CallStatus2) => {
  CallStatus2["STARTED"] = "started";
  CallStatus2["RINGING"] = "ringing";
  CallStatus2["ANSWERED"] = "answered";
  CallStatus2["MACHINE"] = "machine";
  CallStatus2["COMPLETED"] = "completed";
  CallStatus2["BUSY"] = "busy";
  CallStatus2["CANCELLED"] = "cancelled";
  CallStatus2["FAILED"] = "failed";
  CallStatus2["REJECTED"] = "rejected";
  CallStatus2["TIMEOUT"] = "timeout";
  CallStatus2["UNANSWERED"] = "unanswered";
  return CallStatus2;
})(CallStatus || {});

// lib/enums/HttpMethod.ts
var HttpMethod = /* @__PURE__ */ ((HttpMethod2) => {
  HttpMethod2["GET"] = "GET";
  HttpMethod2["POST"] = "POST";
  return HttpMethod2;
})(HttpMethod || {});

// lib/enums/MachineDetctionBehavior.ts
var MachineDetectionBehavior = /* @__PURE__ */ ((MachineDetectionBehavior2) => {
  MachineDetectionBehavior2["CONTINUE"] = "continue";
  MachineDetectionBehavior2["HANGUP"] = "hangup";
  return MachineDetectionBehavior2;
})(MachineDetectionBehavior || {});

// lib/enums/NCCOActions.ts
var NCCOActions = /* @__PURE__ */ ((NCCOActions2) => {
  NCCOActions2["CONNECT"] = "connect";
  NCCOActions2["CONVERSATION"] = "conversation";
  NCCOActions2["INPUT"] = "input";
  NCCOActions2["NOTIFY"] = "notify";
  NCCOActions2["RECORD"] = "record";
  NCCOActions2["STREAM"] = "stream";
  NCCOActions2["TALK"] = "talk";
  return NCCOActions2;
})(NCCOActions || {});

// lib/enums/TTSLanguages.ts
var TTSLanguages = /* @__PURE__ */ ((TTSLanguages2) => {
  TTSLanguages2["AF_ZA"] = "af-ZA";
  TTSLanguages2["AR"] = "ar";
  TTSLanguages2["BG_BG"] = "bg-BG";
  TTSLanguages2["BN_IN"] = "bn-IN";
  TTSLanguages2["CA_ES"] = "ca-ES";
  TTSLanguages2["CMN_CN"] = "cmn-CN";
  TTSLanguages2["CMN_TW"] = "cmn-TW";
  TTSLanguages2["CS_CZ"] = "cs-CZ";
  TTSLanguages2["CY_GB"] = "cy-GB";
  TTSLanguages2["DA_DK"] = "da-DK";
  TTSLanguages2["DE_AT"] = "de-AT";
  TTSLanguages2["DE_DE"] = "de-DE";
  TTSLanguages2["DE_ET"] = "de-ET";
  TTSLanguages2["EL_GR"] = "el-GR";
  TTSLanguages2["EN_AU"] = "en-AU";
  TTSLanguages2["EN_GB"] = "en-GB";
  TTSLanguages2["EN_GB_SCT"] = "en-GB-SCT";
  TTSLanguages2["EN_GB_WLS"] = "en-GB-WLS";
  TTSLanguages2["EN_IE"] = "en-IE";
  TTSLanguages2["EN_IN"] = "en-IN";
  TTSLanguages2["EN_NZ"] = "en-NZ";
  TTSLanguages2["EN_US"] = "en-US";
  TTSLanguages2["EN_ZA"] = "en-ZA";
  TTSLanguages2["ES_CO"] = "es-CO";
  TTSLanguages2["ES_ES"] = "es-ES";
  TTSLanguages2["ES_MX"] = "es-MX";
  TTSLanguages2["ES_US"] = "es-US";
  TTSLanguages2["EU_ES"] = "eu-ES";
  TTSLanguages2["FI_FI"] = "fi-FI";
  TTSLanguages2["FIL_PH"] = "fil-PH";
  TTSLanguages2["FR_CA"] = "fr-CA";
  TTSLanguages2["FR_FR"] = "fr-FR";
  TTSLanguages2["GL_ES"] = "gl-ES";
  TTSLanguages2["GU_IN"] = "gu-IN";
  TTSLanguages2["HE_IL"] = "he-IL";
  TTSLanguages2["HI_IN"] = "hi-IN";
  TTSLanguages2["HU_HU"] = "hu-HU";
  TTSLanguages2["ID_ID"] = "id-ID";
  TTSLanguages2["IS_IS"] = "is-IS";
  TTSLanguages2["IT_IT"] = "it-IT";
  TTSLanguages2["JA_JP"] = "ja-JP";
  TTSLanguages2["KN_IN"] = "kn-IN";
  TTSLanguages2["KO_KR"] = "ko-KR";
  TTSLanguages2["LV_LV"] = "lv-LV";
  TTSLanguages2["ML_IN"] = "ml-IN";
  TTSLanguages2["MS_MY"] = "ms-MY";
  TTSLanguages2["NB_NO"] = "nb-NO";
  TTSLanguages2["NL_BE"] = "nl-BE";
  TTSLanguages2["NL_NL"] = "nl-NL";
  TTSLanguages2["NO_NO"] = "no-NO";
  TTSLanguages2["PA_IN"] = "pa-IN";
  TTSLanguages2["PL_PL"] = "pl-PL";
  TTSLanguages2["PT_BR"] = "pt-BR";
  TTSLanguages2["PT_PT"] = "pt-PT";
  TTSLanguages2["RO_RO"] = "ro-RO";
  TTSLanguages2["RU_RU"] = "ru-RU";
  TTSLanguages2["SK_SK"] = "sk-SK";
  TTSLanguages2["SR_RS"] = "sr-RS";
  TTSLanguages2["SV_SE"] = "sv-SE";
  TTSLanguages2["TA_IN"] = "ta-IN";
  TTSLanguages2["TE_IN"] = "te-IN";
  TTSLanguages2["TH_TH"] = "th-TH";
  TTSLanguages2["TR_TR"] = "tr-TR";
  TTSLanguages2["UK_UA"] = "uk-UA";
  TTSLanguages2["VI_VN"] = "vi-VN";
  TTSLanguages2["YUE_CN"] = "yue-CN";
  return TTSLanguages2;
})(TTSLanguages || {});

// lib/enums/TTSVoices.ts
var TTSVoices = /* @__PURE__ */ ((TTSVoices2) => {
  TTSVoices2["ADITI"] = "Aditi";
  TTSVoices2["AGNIESZKA"] = "Agnieszka";
  TTSVoices2["ALVA"] = "Alva";
  TTSVoices2["AMY"] = "Amy";
  TTSVoices2["ASTRID"] = "Astrid";
  TTSVoices2["BIANCA"] = "Bianca";
  TTSVoices2["BRIAN"] = "Brian";
  TTSVoices2["CARLA"] = "Carla";
  TTSVoices2["CARMEN"] = "Carmen";
  TTSVoices2["CARMIT"] = "Carmit";
  TTSVoices2["CATARINA"] = "Catarina";
  TTSVoices2["CELINE"] = "Celine";
  TTSVoices2["CEM"] = "Cem";
  TTSVoices2["CHANTAL"] = "Chantal";
  TTSVoices2["CHIPMUNK"] = "Chipmunk";
  TTSVoices2["CONCHITA"] = "Conchita";
  TTSVoices2["CRISTIANO"] = "Cristiano";
  TTSVoices2["DAMAYANTI"] = "Damayanti";
  TTSVoices2["DORA"] = "Dora";
  TTSVoices2["EMMA"] = "Emma";
  TTSVoices2["EMPAR"] = "Empar";
  TTSVoices2["ENRIQUE"] = "Enrique";
  TTSVoices2["ERIC"] = "Eric";
  TTSVoices2["EWA"] = "Ewa";
  TTSVoices2["FELIPE"] = "Felipe";
  TTSVoices2["FILIZ"] = "Filiz";
  TTSVoices2["GERAINT"] = "Geraint";
  TTSVoices2["GIORGIO"] = "Giorgio";
  TTSVoices2["GWYNETH"] = "Gwyneth";
  TTSVoices2["HANS"] = "Hans";
  TTSVoices2["HENRIK"] = "Henrik";
  TTSVoices2["INES"] = "Ines";
  TTSVoices2["IOANA"] = "Ioana";
  TTSVoices2["IVETA"] = "Iveta";
  TTSVoices2["IVY"] = "Ivy";
  TTSVoices2["JACEK"] = "Jacek";
  TTSVoices2["JAN"] = "Jan";
  TTSVoices2["JENNIFER"] = "Jennifer";
  TTSVoices2["JOANA"] = "Joana";
  TTSVoices2["JOANNA"] = "Joanna";
  TTSVoices2["JOEY"] = "Joey";
  TTSVoices2["JORDI"] = "Jordi";
  TTSVoices2["JUSTIN"] = "Justin";
  TTSVoices2["KANYA"] = "Kanya";
  TTSVoices2["KARL"] = "Karl";
  TTSVoices2["KENDRA"] = "Kendra";
  TTSVoices2["KIMBERLY"] = "Kimberly";
  TTSVoices2["LAILA"] = "Laila";
  TTSVoices2["LAURA"] = "Laura";
  TTSVoices2["LEA"] = "Lea";
  TTSVoices2["LEKHA"] = "Lekha";
  TTSVoices2["LIV"] = "Liv";
  TTSVoices2["LOTTE"] = "Lotte";
  TTSVoices2["LUCIA"] = "Lucia";
  TTSVoices2["LUCIANA"] = "Luciana";
  TTSVoices2["MADS"] = "Mads";
  TTSVoices2["MAGED"] = "Maged";
  TTSVoices2["MAJA"] = "Maja";
  TTSVoices2["MARISKA"] = "Mariska";
  TTSVoices2["MARLENE"] = "Marlene";
  TTSVoices2["MATHIEU"] = "Mathieu";
  TTSVoices2["MATTHEW"] = "Matthew";
  TTSVoices2["MAXIM"] = "Maxim";
  TTSVoices2["MEI_JIA"] = "Mei-Jia";
  TTSVoices2["MELINA"] = "Melina";
  TTSVoices2["MIA"] = "Mia";
  TTSVoices2["MIGUEL"] = "Miguel";
  TTSVoices2["MIREN"] = "Miren";
  TTSVoices2["MIZUKI"] = "Mizuki";
  TTSVoices2["MONTSERRAT"] = "Montserrat";
  TTSVoices2["NAJA"] = "Naja";
  TTSVoices2["NICOLE"] = "Nicole";
  TTSVoices2["NIKOS"] = "Nikos";
  TTSVoices2["NORA"] = "Nora";
  TTSVoices2["OSKAR"] = "Oskar";
  TTSVoices2["PENELOPE"] = "Penelope";
  TTSVoices2["RAVEENA"] = "Raveena";
  TTSVoices2["RICARDO"] = "Ricardo";
  TTSVoices2["RUBEN"] = "Ruben";
  TTSVoices2["RUSSELL"] = "Russell";
  TTSVoices2["SALLI"] = "Salli";
  TTSVoices2["SATU"] = "Satu";
  TTSVoices2["SEOYEON"] = "Seoyeon";
  TTSVoices2["SIN_JI"] = "Sin-Ji";
  TTSVoices2["SORA"] = "Sora";
  TTSVoices2["TAKUMI"] = "Takumi";
  TTSVoices2["TARIK"] = "Tarik";
  TTSVoices2["TATYANA"] = "Tatyana";
  TTSVoices2["TESSA"] = "Tessa";
  TTSVoices2["TIAN_TIAN"] = "Tian-Tian";
  TTSVoices2["VICKI"] = "Vicki";
  TTSVoices2["VITORIA"] = "Vitoria";
  TTSVoices2["YELDA"] = "Yelda";
  TTSVoices2["ZEINA"] = "Zeina";
  TTSVoices2["ZHIYU"] = "Zhiyu";
  TTSVoices2["ZUZANA"] = "Zuzana";
  return TTSVoices2;
})(TTSVoices || {});

// lib/enums/Endpoint/WebsocketBitrate.ts
var WebsocketBitrate = /* @__PURE__ */ ((WebsocketBitrate2) => {
  WebsocketBitrate2["RATE_8000"] = "audio/l16;rate=8000";
  WebsocketBitrate2["RATE_16000"] = "audio/l16;rate=16000";
  return WebsocketBitrate2;
})(WebsocketBitrate || {});

// lib/enums/NCCO/ConnectEventType.ts
var ConnectEventType = /* @__PURE__ */ ((ConnectEventType2) => {
  ConnectEventType2["SYNCHRONOUS"] = "synchronous";
  return ConnectEventType2;
})(ConnectEventType || {});

// lib/enums/NCCO/MachineDetection.ts
var MachineDetection = /* @__PURE__ */ ((MachineDetection2) => {
  MachineDetection2["CONTINUE"] = "continue";
  MachineDetection2["HANGUP"] = "hangup";
  return MachineDetection2;
})(MachineDetection || {});

// lib/enums/NCCO/RecordingFormat.ts
var RecordingFormat = /* @__PURE__ */ ((RecordingFormat2) => {
  RecordingFormat2["MP3"] = "mp3";
  RecordingFormat2["WAV"] = "wav";
  RecordingFormat2["OGG"] = "ogg";
  return RecordingFormat2;
})(RecordingFormat || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AdvancedMachineDetectionMode,
  CallDirection,
  CallStatus,
  ConnectEventType,
  HttpMethod,
  MachineDetection,
  MachineDetectionBehavior,
  NCCOActions,
  RecordingFormat,
  TTSLanguages,
  TTSVoices,
  WebsocketBitrate
});
