// @vonage/voice/dist/esm/enums/index.mjs
import "../chunk-V66WSPX3.mjs";
import "../chunk-RNTVM7DX.mjs";
import {
  MachineDetection
} from "../chunk-3PSD3IGD.mjs";
import {
  RecordingFormat
} from "../chunk-ETZTGOV6.mjs";
import "../chunk-RFGO5BC2.mjs";
import {
  WebsocketBitrate
} from "../chunk-IXODLYK5.mjs";
import {
  ConnectEventType
} from "../chunk-GMDYX2A5.mjs";
import {
  CallDirection
} from "../chunk-NVSQ62IY.mjs";
import {
  CallStatus
} from "../chunk-6UJQTCSN.mjs";
import {
  HttpMethod
} from "../chunk-QE7UENVE.mjs";
import {
  MachineDetectionBehavior
} from "../chunk-3FQRF2AX.mjs";
import {
  NCCOActions
} from "../chunk-BHJ6KK3Z.mjs";
import {
  TTSLanguages
} from "../chunk-Y3T6C6QB.mjs";
import {
  TTSVoices
} from "../chunk-FNFQ27I4.mjs";
import {
  AdvancedMachineDetectionMode
} from "../chunk-NSLUVWNN.mjs";
export {
  AdvancedMachineDetectionMode,
  CallDirection,
  CallStatus,
  ConnectEventType,
  HttpMethod,
  MachineDetection,
  MachineDetectionBehavior,
  NCCOActions,
  RecordingFormat,
  TTSLanguages,
  TTSVoices,
  WebsocketBitrate
};
