// @vonage/redact/dist/cjs/enums/product.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/product.ts
var product_exports = {};
__export(product_exports, {
  ProductType: () => ProductType
});
module.exports = __toCommonJS(product_exports);
var ProductType = /* @__PURE__ */ ((ProductType2) => {
  ProductType2["SMS"] = "SMS";
  ProductType2["Voice"] = "Voice";
  ProductType2["NumberInsight"] = "NumberInsight";
  ProductType2["Verify"] = "Verify";
  ProductType2["VerifySDK"] = "VerifySDK";
  return ProductType2;
})(ProductType || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ProductType
});
