// @vonage/verify/dist/esm/chunk-GOUKMLMF.mjs
// lib/enums/VerifyLanguages.ts
var VerifyLanguages = /* @__PURE__ */ ((VerifyLanguages2) => {
  VerifyLanguages2["ARABIC"] = "ar-xa";
  VerifyLanguages2["CZECH"] = "cs-cz";
  VerifyLanguages2["WELSH"] = "cy-cy";
  VerifyLanguages2["WELSH_UK"] = "cy-gb";
  VerifyLanguages2["DANISH"] = "da-dk";
  VerifyLanguages2["GERMAN"] = "de-de";
  VerifyLanguages2["GREEK"] = "el-gr";
  VerifyLanguages2["ENGLISH_AUSTRALIA"] = "en-au";
  VerifyLanguages2["ENGLISH_UK"] = "en-gb";
  VerifyLanguages2["ENGLISH_INDIA"] = "en-in";
  VerifyLanguages2["ENGLISH_US"] = "en-us";
  VerifyLanguages2["SPANISH_SPAIN"] = "es-es";
  VerifyLanguages2["SPANISH_MEXICO"] = "es-mx";
  VerifyLanguages2["SPANISH_US"] = "es-us";
  VerifyLanguages2["FINNISH"] = "fi-fi";
  VerifyLanguages2["FILIPINO"] = "fil-ph";
  VerifyLanguages2["FRENCH_CANADA"] = "fr-ca";
  VerifyLanguages2["FRENCH_FRANCE"] = "fr-fr";
  VerifyLanguages2["HINDI"] = "hi-in";
  VerifyLanguages2["HUNGARIAN"] = "hu-hu";
  VerifyLanguages2["INDONESIAN"] = "id-id";
  VerifyLanguages2["ICELANDIC"] = "is-is";
  VerifyLanguages2["ITALIAN"] = "it-it";
  VerifyLanguages2["JAPANESE"] = "ja-jp";
  VerifyLanguages2["KOREAN"] = "ko-kr";
  VerifyLanguages2["NORWEGIAN"] = "nb-no";
  VerifyLanguages2["DUTCH"] = "nl-nl";
  VerifyLanguages2["POLISH"] = "pl-pl";
  VerifyLanguages2["PORTUGUESE_BRAZIL"] = "pt-br";
  VerifyLanguages2["PORTUGUESE_PORTUGAL"] = "pt-pt";
  VerifyLanguages2["ROMANIAN"] = "ro-ro";
  VerifyLanguages2["RUSSIAN"] = "ru-ru";
  VerifyLanguages2["SWEDISH"] = "sv-se";
  VerifyLanguages2["THAI"] = "th-th";
  VerifyLanguages2["TURKISH"] = "tr-tr";
  VerifyLanguages2["VIETNAMESE"] = "vi-vn";
  VerifyLanguages2["YUE_CHINESE_CANTONESE"] = "yue-cn";
  VerifyLanguages2["CHINESE_MAINLAND"] = "zh-cn";
  VerifyLanguages2["CHINESE_TAIWAN"] = "zh-tw";
  return VerifyLanguages2;
})(VerifyLanguages || {});

export {
  VerifyLanguages
};
