// @vonage/server-client/dist/esm/chunk-LIWWKYYJ.mjs
// lib/enums/GenericErrorCodes.ts
var GenericErrors = /* @__PURE__ */ ((GenericErrors2) => {
  GenericErrors2["LOW_BALANCE"] = "low-balance";
  GenericErrors2["UNAUTHORIZED"] = "unauthorized";
  GenericErrors2["FORBIDDEN"] = "forbidden";
  GenericErrors2["NOT_FOUND"] = "not-found";
  GenericErrors2["UNPROVISIONED"] = "unprovisioned";
  GenericErrors2["ACCOUNT_SUSPENDED"] = "account-suspended";
  GenericErrors2["JWT_EXPIRED"] = "jwt-expired";
  GenericErrors2["JWT_REVOKED"] = "jwt-revoked";
  GenericErrors2["INVALID_API_KEY"] = "invalid-api-key";
  GenericErrors2["INVALID_SIGNATURE"] = "invalid-signature";
  GenericErrors2["INVALID_IP"] = "invalid-ip";
  GenericErrors2["MULTIPLE_AUTH_METHODS"] = "multiple-auth-methods";
  GenericErrors2["INVALID_ID"] = "invalid-id";
  GenericErrors2["INVALID_JSON"] = "invalid-json";
  GenericErrors2["WRONG_VERB"] = "wrong-verb";
  GenericErrors2["ACCEPT_HEADER"] = "accept-header";
  GenericErrors2["CONTENT_TYPE_HEADER"] = "content-type-header";
  GenericErrors2["UNAVAILABLE_LEGAL"] = "unavailable-legal";
  GenericErrors2["APPLICATION_SUSPENDED"] = "application-suspended";
  return GenericErrors2;
})(GenericErrors || {});

export {
  GenericErrors
};
