// @vonage/sms/dist/esm/enums/SMSErrors.mjs
// lib/enums/SMSErrors.ts
var SMSErrors = /* @__PURE__ */ ((SMSErrors2) => {
  SMSErrors2["SUCCESS"] = "0";
  SMSErrors2["THROTTLED"] = "1";
  SMSErrors2["MISSING_REQUIRED_PARAM"] = "2";
  SMSErrors2["INVALID_PARAM"] = "3";
  SMSErrors2["INVALID_CREDENTIALS"] = "4";
  SMSErrors2["INTERNAL_ERROR"] = "5";
  SMSErrors2["INVALID_MESSAGE"] = "6";
  SMSErrors2["BARRED_NUMBER"] = "7";
  SMSErrors2["PARTNER_ACCOUNT_BARRED"] = "8";
  SMSErrors2["PARTNER_QUOTA_VIOLATION"] = "9";
  SMSErrors2["TOO_MANY_EXISTING_BINDS"] = "10";
  SMSErrors2["ACCOUNT_NOT_ENABLED"] = "11";
  SMSErrors2["MESSAGE_TOO_LONG"] = "12";
  SMSErrors2["INVALID_SIGNATURE"] = "14";
  SMSErrors2["INVALID_SENDER_ADDRESS"] = "15";
  SMSErrors2["INVALID_NETWORK_CODE"] = "22";
  SMSErrors2["INVALID_CALLBACK_URL"] = "23";
  SMSErrors2["NON_WHITELISTED_DESTINATION"] = "29";
  SMSErrors2["SIGNATURE_AND_API_SECRET_DISALLOWED"] = "32";
  SMSErrors2["NUMBER_DEACTIVATED"] = "33";
  return SMSErrors2;
})(SMSErrors || {});
export {
  SMSErrors
};
