// @vonage/verify2/dist/cjs/enums/channels.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/channels.ts
var channels_exports = {};
__export(channels_exports, {
  Channels: () => Channels,
  SilentAuthChannel: () => SilentAuthChannel
});
module.exports = __toCommonJS(channels_exports);
var Channels = /* @__PURE__ */ ((Channels2) => {
  Channels2["SMS"] = "sms";
  Channels2["WHATSAPP"] = "whatsapp";
  Channels2["WHATSAPP_INTERACTIVE"] = "whatsapp_interactive";
  Channels2["VOICE"] = "voice";
  Channels2["RCS"] = "rcs";
  Channels2["EMAIL"] = "email";
  return Channels2;
})(Channels || {});
var SilentAuthChannel = /* @__PURE__ */ ((SilentAuthChannel2) => {
  SilentAuthChannel2["SILENT_AUTH"] = "silent_auth";
  return SilentAuthChannel2;
})(SilentAuthChannel || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Channels,
  SilentAuthChannel
});
