// @vonage/verify2/dist/cjs/enums/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/index.ts
var enums_exports = {};
__export(enums_exports, {
  Channels: () => Channels,
  EventStatus: () => EventStatus,
  RequestStatus: () => RequestStatus,
  RequestWorkflowStatus: () => RequestWorkflowStatus,
  SilentAuthChannel: () => SilentAuthChannel,
  SilentAuthStatus: () => SilentAuthStatus,
  VerifyLocale: () => VerifyLocale
});
module.exports = __toCommonJS(enums_exports);

// lib/enums/channels.ts
var Channels = /* @__PURE__ */ ((Channels2) => {
  Channels2["SMS"] = "sms";
  Channels2["WHATSAPP"] = "whatsapp";
  Channels2["WHATSAPP_INTERACTIVE"] = "whatsapp_interactive";
  Channels2["VOICE"] = "voice";
  Channels2["RCS"] = "rcs";
  Channels2["EMAIL"] = "email";
  return Channels2;
})(Channels || {});
var SilentAuthChannel = /* @__PURE__ */ ((SilentAuthChannel2) => {
  SilentAuthChannel2["SILENT_AUTH"] = "silent_auth";
  return SilentAuthChannel2;
})(SilentAuthChannel || {});

// lib/enums/eventStatus.ts
var EventStatus = /* @__PURE__ */ ((EventStatus2) => {
  EventStatus2["COMPLETED"] = "completed";
  EventStatus2["FAILED"] = "failed";
  EventStatus2["USER_REJECTED"] = "user_rejected";
  EventStatus2["REJECTED"] = "rejected";
  return EventStatus2;
})(EventStatus || {});

// lib/enums/requestStatus.ts
var RequestStatus = /* @__PURE__ */ ((RequestStatus2) => {
  RequestStatus2["COMPLETED"] = "completed";
  RequestStatus2["FAILED"] = "failed";
  RequestStatus2["EXPIRED"] = "expired";
  return RequestStatus2;
})(RequestStatus || {});

// lib/enums/requestWorkflowsStatus.ts
var RequestWorkflowStatus = /* @__PURE__ */ ((RequestWorkflowStatus2) => {
  RequestWorkflowStatus2["UNUSED"] = "unused";
  RequestWorkflowStatus2["COMPLETED"] = "completed";
  RequestWorkflowStatus2["FAILED"] = "failed";
  RequestWorkflowStatus2["EXPIRED"] = "expired";
  RequestWorkflowStatus2["USER_REJECTED"] = "user_rejected";
  return RequestWorkflowStatus2;
})(RequestWorkflowStatus || {});

// lib/enums/silentAuthStatus.ts
var SilentAuthStatus = /* @__PURE__ */ ((SilentAuthStatus2) => {
  SilentAuthStatus2["COMPLETED"] = "completed";
  SilentAuthStatus2["FAILED"] = "failed";
  SilentAuthStatus2["USER_REJECTED"] = "user_rejected";
  SilentAuthStatus2["EXPIRED"] = "expired";
  SilentAuthStatus2["ACTION_PENDING"] = "action_pending";
  return SilentAuthStatus2;
})(SilentAuthStatus || {});

// lib/enums/verifyLocale.ts
var VerifyLocale = /* @__PURE__ */ ((VerifyLocale2) => {
  VerifyLocale2["EN_US"] = "en-us";
  VerifyLocale2["EN_GB"] = "en-gb";
  VerifyLocale2["ES_ES"] = "es-es";
  VerifyLocale2["ES_MX"] = "es-mx";
  VerifyLocale2["ES_US"] = "es-us";
  VerifyLocale2["IT_IT"] = "it-it";
  VerifyLocale2["FR_FR"] = "fr-fr";
  VerifyLocale2["DE_DE"] = "de-de";
  VerifyLocale2["RU_RU"] = "ru-ru";
  VerifyLocale2["HI_IN"] = "hi-in";
  VerifyLocale2["PT_BR"] = "pt-br";
  VerifyLocale2["PT_PT"] = "pt-pt";
  VerifyLocale2["ID_ID"] = "id-id";
  return VerifyLocale2;
})(VerifyLocale || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Channels,
  EventStatus,
  RequestStatus,
  RequestWorkflowStatus,
  SilentAuthChannel,
  SilentAuthStatus,
  VerifyLocale
});
