// @vonage/verify2/dist/cjs/enums/silentAuthStatus.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/silentAuthStatus.ts
var silentAuthStatus_exports = {};
__export(silentAuthStatus_exports, {
  SilentAuthStatus: () => SilentAuthStatus
});
module.exports = __toCommonJS(silentAuthStatus_exports);
var SilentAuthStatus = /* @__PURE__ */ ((SilentAuthStatus2) => {
  SilentAuthStatus2["COMPLETED"] = "completed";
  SilentAuthStatus2["FAILED"] = "failed";
  SilentAuthStatus2["USER_REJECTED"] = "user_rejected";
  SilentAuthStatus2["EXPIRED"] = "expired";
  SilentAuthStatus2["ACTION_PENDING"] = "action_pending";
  return SilentAuthStatus2;
})(SilentAuthStatus || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SilentAuthStatus
});
