// @vonage/verify2/dist/cjs/enums/verifyLocale.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/verifyLocale.ts
var verifyLocale_exports = {};
__export(verifyLocale_exports, {
  VerifyLocale: () => VerifyLocale
});
module.exports = __toCommonJS(verifyLocale_exports);
var VerifyLocale = /* @__PURE__ */ ((VerifyLocale2) => {
  VerifyLocale2["EN_US"] = "en-us";
  VerifyLocale2["EN_GB"] = "en-gb";
  VerifyLocale2["ES_ES"] = "es-es";
  VerifyLocale2["ES_MX"] = "es-mx";
  VerifyLocale2["ES_US"] = "es-us";
  VerifyLocale2["IT_IT"] = "it-it";
  VerifyLocale2["FR_FR"] = "fr-fr";
  VerifyLocale2["DE_DE"] = "de-de";
  VerifyLocale2["RU_RU"] = "ru-ru";
  VerifyLocale2["HI_IN"] = "hi-in";
  VerifyLocale2["PT_BR"] = "pt-br";
  VerifyLocale2["PT_PT"] = "pt-pt";
  VerifyLocale2["ID_ID"] = "id-id";
  return VerifyLocale2;
})(VerifyLocale || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  VerifyLocale
});
