// @vonage/verify/dist/cjs/classes/PSD2.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/classes/PSD2.ts
var PSD2_exports = {};
__export(PSD2_exports, {
  PSD2: () => PSD2
});
module.exports = __toCommonJS(PSD2_exports);
var PSD2 = class {
  /**
   * The phone number to be verified.
   */
  number;
  /**
   * The payee's name or identifier for the payment confirmation.
   */
  payee;
  /**
   * The decimal amount of the payment to be confirmed, in Euros.
   */
  amount;
  /**
   * The country code associated with the phone number (optional).
   */
  country;
  /**
   * The desired length of the verification code (optional).
   */
  codeLength;
  /**
   * @deprecated This property is deprecated. Use `lg` instead.
   */
  language;
  /**
   * The language for sending verification messages (optional).
   */
  lg;
  /**
   * The duration in seconds for which the verification code will be valid.
   */
  pinExpiry;
  /**
   * The duration in seconds to wait before sending the next verification event.
   */
  nextEventWait;
  /**
   * The workflow ID for customizing the verification process (optional).
   */
  workflowId;
  constructor(phoneNumber, payee, amount, country, codeLength, language, lg, pinExpiry, nextEventWait, workflowId) {
    this.number = phoneNumber;
    this.amount = amount;
    this.payee = payee;
    if (country) {
      this.country = country;
    }
    if (codeLength) {
      this.codeLength = codeLength;
    }
    if (language) {
      this.lg = language;
      this.language = language;
    }
    if (pinExpiry) {
      this.pinExpiry = pinExpiry;
    }
    if (nextEventWait) {
      this.nextEventWait = nextEventWait;
    }
    if (workflowId) {
      this.workflowId = workflowId;
    }
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PSD2
});
