// @vonage/users/dist/cjs/user.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/user.ts
var user_exports = {};
__export(user_exports, {
  Users: () => Users
});
module.exports = __toCommonJS(user_exports);
var import_server_client = require("@vonage/server-client");
var apiUserToUser = (apiUser) => {
  delete apiUser._links;
  const user = import_server_client.Client.transformers.camelCaseObjectKeys(
    apiUser,
    true
  );
  if (user.properties) {
    user.properties.customData = apiUser.properties.custom_data;
  }
  if (user.channels?.websocket) {
    user.channels.websocket = apiUser.channels.websocket.map(
      (apiSocket) => {
        const socket = {
          uri: apiSocket.uri
        };
        if (apiSocket["content-type"]) {
          socket.contentType = apiSocket["content-type"];
        }
        if (apiSocket.headers) {
          socket.headers = apiSocket.headers;
        }
        return socket;
      }
    );
  }
  return user;
};
var userToAPI = (user) => {
  const apiUser = import_server_client.Client.transformers.snakeCaseObjectKeys(
    user,
    true
  );
  if (user.properties?.customData) {
    apiUser.properties.custom_data = user.properties.customData;
  }
  if (user.channels?.websocket) {
    apiUser.channels.websocket = user.channels.websocket?.map(
      (socket) => {
        const apiSocket = {
          uri: socket.uri
        };
        if (socket.contentType) {
          apiSocket["content-type"] = socket.contentType;
        }
        if (socket.headers) {
          apiSocket.headers = socket.headers;
        }
        return apiSocket;
      }
    );
  }
  delete apiUser.id;
  return apiUser;
};
var Users = class extends import_server_client.Client {
  authType = import_server_client.AuthenticationType.JWT;
  /**
   * Retrieves a list of users, optionally paginated, based on the provided parameters.
   *
   * @param {UserListParameters} params - Optional parameters to filter and paginate the list of users.
   * @return {AsyncGenerator<UserType, void, undefined>} An async generator that yields user objects.
   */
  async *listAllUsers(params = {}) {
    let cursor = params.cursor;
    do {
      if (cursor) {
        params.cursor = cursor;
      }
      const resp = await this.getUserPage(params);
      yield* resp._embedded?.users.map(apiUserToUser);
      const next = resp._links?.next?.href ? new URL(resp._links.next.href) : null;
      cursor = next ? `${next.searchParams.get("cursor")}` : void 0;
    } while (cursor);
  }
  /**
   * Retrieves a page of users based on the provided parameters, such as pagination and filtering.
   *
   * @param {UserListParameters} [params={}] - Optional parameters to filter and paginate the list of users.
   * @param {number} [params.pageSize] - The number of users to include per page.
   * @param {SortOrder} [params.order] - The sorting order for the list (ASC or DESC).
   * @param {string} [params.cursor] - A cursor for paginating through the user list.
   * @param {string} [params.name] - A name to filter users by.
   *
   * @return {Promise<UserPageResponse>} A Promise that resolves to a UserPageResponse object containing the user page data.
   *
   * @throws {Error} If there is an issue with the request or response.
   */
  async getUserPage(params = {}) {
    const resp = await this.sendGetRequest(
      `${this.config.apiHost}/v1/users`,
      import_server_client.Client.transformers.snakeCaseObjectKeys(params, true)
    );
    return resp.data;
  }
  /**
   * Creates a new user with the provided user data.
   *
   * @param {UserType} user - The user data to create a new user.
   * @return {Promise<UserType>} A Promise that resolves to the newly created user.
   * @throws {Error} If there is an issue with the request or response.
   */
  async createUser(user) {
    const resp = await this.sendPostRequest(
      `${this.config.apiHost}/v1/users`,
      userToAPI(user)
    );
    return apiUserToUser(resp.data);
  }
  /**
   * Retrieves user information for the specified user ID.
   *
   * @param {string} userId - The unique identifier of the user to retrieve.
   * @return {Promise<UserType>} A Promise that resolves to the user information for the specified user ID.
   * @throws {Error} If there is an issue with the request or response, or if the user with the specified ID is not found.
   */
  async getUser(userId) {
    const resp = await this.sendGetRequest(
      `${this.config.apiHost}/v1/users/${userId}`
    );
    return apiUserToUser(resp.data);
  }
  /**
   * Updates the user information for the specified user.
   *
   * @param {UserType} user - The user object containing the updated information.
   * @return {Promise<UserType>} A Promise that resolves to the updated user information.
   * @throws {Error} If there is an issue with the request or response, or if the user with the specified ID is not found.
   */
  async updateUser(user) {
    const resp = await this.sendPatchRequest(
      `${this.config.apiHost}/v1/users/${user.id}`,
      userToAPI(user)
    );
    return apiUserToUser(resp.data);
  }
  /**
   * Deletes the user with the specified user ID.
   *
   * @param {string} userId - The unique ID of the user to be deleted.
   * @return {Promise<void>} A Promise that resolves once the user is successfully deleted.
   * @throws {Error} If there is an issue with the request or response, or if the user with the specified ID is not found.
   */
  async deleteUser(userId) {
    await this.sendDeleteRequest(
      `${this.config.apiHost}/v1/users/${userId}`
    );
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Users
});
