// @vonage/auth/dist/cjs/enums/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/index.ts
var enums_exports = {};
__export(enums_exports, {
  AlgorithmTypes: () => AlgorithmTypes
});
module.exports = __toCommonJS(enums_exports);

// lib/enums/AlgroithmTypes.ts
var AlgorithmTypes = /* @__PURE__ */ ((AlgorithmTypes2) => {
  AlgorithmTypes2["md5hash"] = "MD5HASH";
  AlgorithmTypes2["md5hmac"] = "MD5HMAC";
  AlgorithmTypes2["sha1hmac"] = "SHA1HMAC";
  AlgorithmTypes2["sha256hmac"] = "SHA256HMAC";
  AlgorithmTypes2["sha512hmac"] = "SHA512HMAC";
  return AlgorithmTypes2;
})(AlgorithmTypes || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AlgorithmTypes
});
