// @vonage/numbers/dist/cjs/enums/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/index.ts
var enums_exports = {};
__export(enums_exports, {
  Feature: () => Feature,
  LineType: () => LineType,
  MessagesCallbackTypeEnum: () => MessagesCallbackTypeEnum,
  SearchPattern: () => SearchPattern,
  VoiceCallbackTypeEnum: () => VoiceCallbackTypeEnum
});
module.exports = __toCommonJS(enums_exports);

// lib/enums/Feature.ts
var Feature = /* @__PURE__ */ ((Feature2) => {
  Feature2["MMS"] = "MMS";
  Feature2["SMS"] = "SMS";
  Feature2["VOICE"] = "VOICE";
  return Feature2;
})(Feature || {});

// lib/enums/LineType.ts
var LineType = /* @__PURE__ */ ((LineType2) => {
  LineType2["LANDLINE"] = "landline";
  LineType2["MOBILE_LVN"] = "mobile-lvn";
  LineType2["LANDLINE_TOLL_FREE"] = "landline-toll-free";
  return LineType2;
})(LineType || {});

// lib/enums/MessagesCallbackTypeEnum.ts
var MessagesCallbackTypeEnum = /* @__PURE__ */ ((MessagesCallbackTypeEnum2) => {
  MessagesCallbackTypeEnum2["App"] = "app";
  return MessagesCallbackTypeEnum2;
})(MessagesCallbackTypeEnum || {});

// lib/enums/SearchPattern.ts
var SearchPattern = /* @__PURE__ */ ((SearchPattern2) => {
  SearchPattern2[SearchPattern2["START_WITH"] = 0] = "START_WITH";
  SearchPattern2[SearchPattern2["CONTAINS"] = 1] = "CONTAINS";
  SearchPattern2[SearchPattern2["ENDS_WITH"] = 2] = "ENDS_WITH";
  return SearchPattern2;
})(SearchPattern || {});

// lib/enums/VoiceCallbackTypeEnum.ts
var VoiceCallbackTypeEnum = /* @__PURE__ */ ((VoiceCallbackTypeEnum2) => {
  VoiceCallbackTypeEnum2["Sip"] = "sip";
  VoiceCallbackTypeEnum2["Tel"] = "tel";
  VoiceCallbackTypeEnum2["App"] = "app";
  return VoiceCallbackTypeEnum2;
})(VoiceCallbackTypeEnum || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Feature,
  LineType,
  MessagesCallbackTypeEnum,
  SearchPattern,
  VoiceCallbackTypeEnum
});
