// @vonage/video/dist/cjs/enums/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/index.ts
var enums_exports = {};
__export(enums_exports, {
  ArchiveOutputMode: () => ArchiveOutputMode,
  AudioRate: () => AudioRate,
  CaptionStatus: () => CaptionStatus,
  ExperienceComposerResolution: () => ExperienceComposerResolution,
  LayoutType: () => LayoutType,
  MediaMode: () => MediaMode,
  Resolution: () => Resolution,
  StreamMode: () => StreamMode
});
module.exports = __toCommonJS(enums_exports);

// lib/enums/ArchiveOutputMode.ts
var ArchiveOutputMode = /* @__PURE__ */ ((ArchiveOutputMode2) => {
  ArchiveOutputMode2["COMPOSED"] = "composed";
  ArchiveOutputMode2["INDIVIDUAL"] = "individual";
  return ArchiveOutputMode2;
})(ArchiveOutputMode || {});

// lib/enums/AudioRate.ts
var AudioRate = /* @__PURE__ */ ((AudioRate2) => {
  AudioRate2[AudioRate2["16KHZ"] = 16e3] = "16KHZ";
  AudioRate2[AudioRate2["8KHZ"] = 8e3] = "8KHZ";
  return AudioRate2;
})(AudioRate || {});

// lib/enums/CaptionStatus.ts
var CaptionStatus = /* @__PURE__ */ ((CaptionStatus2) => {
  CaptionStatus2["STARTED"] = "started";
  CaptionStatus2["STOPPED"] = "stopped";
  CaptionStatus2["PAUSED"] = "paused";
  CaptionStatus2["FAILED"] = "failed";
  return CaptionStatus2;
})(CaptionStatus || {});

// lib/enums/Resolution.ts
var Resolution = /* @__PURE__ */ ((Resolution2) => {
  Resolution2["FHD_LANDSCAPE"] = "1920x1080";
  Resolution2["FHD_PORTRAIT"] = "1080x1920";
  Resolution2["HD_LANDSCAPE"] = "1280x720";
  Resolution2["HD_PORTRAIT"] = "720x1280";
  Resolution2["SD_LANDSCAPE"] = "640x480";
  Resolution2["SD_PORTRAIT"] = "480x640";
  return Resolution2;
})(Resolution || {});

// lib/enums/ExperienceComposerResolution.ts
var ExperienceComposerResolution = ((ExperienceComposerResolution2) => {
  ExperienceComposerResolution2[ExperienceComposerResolution2["SD_LANDSCAPE"] = "640x480" /* SD_LANDSCAPE */] = "SD_LANDSCAPE";
  ExperienceComposerResolution2[ExperienceComposerResolution2["SD_PORTRAIT"] = "480x640" /* SD_PORTRAIT */] = "SD_PORTRAIT";
  ExperienceComposerResolution2[ExperienceComposerResolution2["HD_LANDSCAPE"] = "1280x720" /* HD_LANDSCAPE */] = "HD_LANDSCAPE";
  ExperienceComposerResolution2[ExperienceComposerResolution2["HD_PORTRAIT"] = "720x1280" /* HD_PORTRAIT */] = "HD_PORTRAIT";
  return ExperienceComposerResolution2;
})(ExperienceComposerResolution || {});

// lib/enums/LayoutType.ts
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["BEST_FIT"] = "bestFit";
  LayoutType2["CUSTOM"] = "custom";
  LayoutType2["HORIZONTAL_PRESENTATION"] = "horizontalPresentation";
  LayoutType2["PIP"] = "pip";
  LayoutType2["VERTICAL_PRESENTATION"] = "verticalPresentation";
  return LayoutType2;
})(LayoutType || {});

// lib/enums/StreamMode.ts
var StreamMode = /* @__PURE__ */ ((StreamMode2) => {
  StreamMode2["AUTO"] = "auto";
  StreamMode2["MANUAL"] = "manual";
  return StreamMode2;
})(StreamMode || {});

// lib/enums/MediaMode.ts
var MediaMode = /* @__PURE__ */ ((MediaMode2) => {
  MediaMode2["ROUTED"] = "disabled";
  MediaMode2["RELAYED"] = "enabled";
  return MediaMode2;
})(MediaMode || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ArchiveOutputMode,
  AudioRate,
  CaptionStatus,
  ExperienceComposerResolution,
  LayoutType,
  MediaMode,
  Resolution,
  StreamMode
});
