// @vonage/video/dist/cjs/enums/LayoutType.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/LayoutType.ts
var LayoutType_exports = {};
__export(LayoutType_exports, {
  LayoutType: () => LayoutType
});
module.exports = __toCommonJS(LayoutType_exports);
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["BEST_FIT"] = "bestFit";
  LayoutType2["CUSTOM"] = "custom";
  LayoutType2["HORIZONTAL_PRESENTATION"] = "horizontalPresentation";
  LayoutType2["PIP"] = "pip";
  LayoutType2["VERTICAL_PRESENTATION"] = "verticalPresentation";
  return LayoutType2;
})(LayoutType || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LayoutType
});
