// @vonage/video/dist/cjs/enums/Resolution.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/Resolution.ts
var Resolution_exports = {};
__export(Resolution_exports, {
  Resolution: () => Resolution
});
module.exports = __toCommonJS(Resolution_exports);
var Resolution = /* @__PURE__ */ ((Resolution2) => {
  Resolution2["FHD_LANDSCAPE"] = "1920x1080";
  Resolution2["FHD_PORTRAIT"] = "1080x1920";
  Resolution2["HD_LANDSCAPE"] = "1280x720";
  Resolution2["HD_PORTRAIT"] = "720x1280";
  Resolution2["SD_LANDSCAPE"] = "640x480";
  Resolution2["SD_PORTRAIT"] = "480x640";
  return Resolution2;
})(Resolution || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Resolution
});
