// @vonage/messages/dist/cjs/classes/Viber/Image.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/classes/Viber/Image.ts
var Image_exports = {};
__export(Image_exports, {
  Image: () => Image
});
module.exports = __toCommonJS(Image_exports);

// lib/classes/AbstractMessage.ts
var AbstractMessage = class {
  /**
   * The recipient of the message.
   */
  to;
  /**
   * The sender of the message.
   */
  from;
  /**
   * An optional client reference for the message.
   */
  clientRef;
  /**
   * Specifies the URL to which Status Webhook messages will be sent for this
   * particular message. Over-rides account-level and application-level Status
   * Webhook url settings on a per-message basis.
   */
  webhookUrl;
  /**
   * Specifies which version of the Messages API will be used to send Status
   * Webhook messages for this particular message. For example, if v0.1 is
   * set, then the JSON body of Status Webhook messages for this message will
   * be sent in Messages v0.1 format. Over-rides account-level and
   * application-level API version settings on a per-message basis.
   */
  webhookVersion;
  /**
   * Constructs a new `AbstractMessage` instance.
   *
   * @param {MessageParams} params - The parameters for creating a message.
   */
  constructor(params) {
    this.to = params.to;
    this.from = params.from;
    this.clientRef = params.clientRef;
    this.webhookUrl = params.webhookUrl;
    this.webhookVersion = params.webhookVersion;
  }
};

// lib/classes/AbstractImageMessage.ts
var AbstractImageMessage = class extends AbstractMessage {
  /**
   * The type of message (always 'image').
   */
  messageType = "image" /* IMAGE */;
  image;
  /**
   * Constructs a new `AbstractImageMessage` instance.
   *
   * @param {MessageParamsImage} params - The parameters for creating an image message.
   */
  constructor(params) {
    super(params);
    this.image = params.image;
  }
};

// lib/classes/Viber/ViberImage.ts
var ViberImage = class extends AbstractImageMessage {
  /**
   * The channel for this message (always 'viber_service').
   */
  channel = "viber_service" /* VIBER */;
  viberService;
  /**
   * Send an image message using the Viber Service channel.
   *
   * @param {ViberImageParams} params - The parameters for the ViberImage message.
   * @example
   * ```ts
   * import { ViberImage } from '@vonage/messages';
   *
   * const { messageUUID } = await messagesClient.send(new ViberImage({
   *  to: TO_NUMBER,
   *  from: FROM_NUMBER,
   *  image: {
   *  url: 'https://my-host.com/my-image.jpg',
   *  },
   *  viberService: {
   *    action: {
   *      url: 'https://my-host.com/my-path',
   *      text: 'My button text',
   *    },
   *  },
   *  clientRef: 'my-personal-reference',
   * }));
   *
   * console.log(`Message sent successfully with UUID ${messageUUID}`);
   * ```
   */
  constructor(params) {
    super(params);
    this.viberService = params.viberService;
  }
};

// lib/classes/Viber/Image.ts
var import_debug = __toESM(require("debug"));
var log = (0, import_debug.default)("vonage:messages:viber");
var Image = class extends ViberImage {
  /**
   * Constructs a new `Image` instance for the Viber channel.
   *
   * @param {ImageObject} image - The image content of the message.
   * @param {string} to - The recipient's Viber ID.
   * @param {string} from - The sender's Viber ID.
   * @param {MessageConfig} viberService - The Viber service configuration.
   * @param {string} clientRef - The client reference for the message.
   */
  constructor(image, to, from, viberService, clientRef) {
    log("Please update to use the ViberImage class instead");
    super({
      image,
      to,
      from,
      clientRef,
      viberService
    });
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Image
});
