// @vonage/messages/dist/esm/chunk-UIMLGEV7.mjs
// lib/enums/Messenger/MessengerTags.ts
var MessengerTags = /* @__PURE__ */ ((MessengerTags2) => {
  MessengerTags2["CONFIRMED_EVENT_UPDATE"] = "CONFIRMED_EVENT_UPDATE";
  MessengerTags2["POST_PURCHASE_UPDATE"] = "POST_PURCHASE_UPDATE";
  MessengerTags2["ACCOUNT_UPDATE"] = "ACCOUNT_UPDATE";
  MessengerTags2["HUMAN_AGENT"] = "HUMAN_AGENT";
  MessengerTags2["SHIPPING_UPDATE"] = "SHIPPING_UPDATE";
  MessengerTags2["RESERVATION_UPDATE"] = "RESERVATION_UPDATE";
  MessengerTags2["ISSUE_RESOLUTION"] = "ISSUE_RESOLUTION";
  MessengerTags2["APPOINTMENT_UPDATE"] = "APPOINTMENT_UPDATE";
  MessengerTags2["GAME_EVENT"] = "GAME_EVENT";
  MessengerTags2["TRANSPORTATION_UPDATE"] = "TRANSPORTATION_UPDATE";
  MessengerTags2["FEATURE_FUNCTIONALITY_UPDATE"] = "FEATURE_FUNCTIONALITY_UPDATE";
  MessengerTags2["TICKET_UPDATE"] = "TICKET_UPDATE";
  MessengerTags2["PAYMENT_UPDATE"] = "PAYMENT_UPDATE";
  MessengerTags2["PERSONAL_FINANCE_UPDATE"] = "PERSONAL_FINANCE_UPDATE";
  return MessengerTags2;
})(MessengerTags || {});

export {
  MessengerTags
};
