// @vonage/server-client/dist/cjs/enums/AuthenticationType.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/AuthenticationType.ts
var AuthenticationType_exports = {};
__export(AuthenticationType_exports, {
  AuthenticationType: () => AuthenticationType
});
module.exports = __toCommonJS(AuthenticationType_exports);
var AuthenticationType = /* @__PURE__ */ ((AuthenticationType2) => {
  AuthenticationType2["BASIC"] = "basic";
  AuthenticationType2["JWT"] = "jwt";
  AuthenticationType2["KEY_SECRET"] = "key_secret";
  AuthenticationType2["QUERY_KEY_SECRET"] = "query_key_secret";
  AuthenticationType2["SIGNATURE"] = "signature";
  AuthenticationType2["CIBA"] = "ciba";
  AuthenticationType2["OAUTH2"] = "oauth2";
  return AuthenticationType2;
})(AuthenticationType || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AuthenticationType
});
