// @vonage/server-client/dist/cjs/enums/GenericErrorCodes.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/GenericErrorCodes.ts
var GenericErrorCodes_exports = {};
__export(GenericErrorCodes_exports, {
  GenericErrors: () => GenericErrors
});
module.exports = __toCommonJS(GenericErrorCodes_exports);
var GenericErrors = /* @__PURE__ */ ((GenericErrors2) => {
  GenericErrors2["LOW_BALANCE"] = "low-balance";
  GenericErrors2["UNAUTHORIZED"] = "unauthorized";
  GenericErrors2["FORBIDDEN"] = "forbidden";
  GenericErrors2["NOT_FOUND"] = "not-found";
  GenericErrors2["UNPROVISIONED"] = "unprovisioned";
  GenericErrors2["ACCOUNT_SUSPENDED"] = "account-suspended";
  GenericErrors2["JWT_EXPIRED"] = "jwt-expired";
  GenericErrors2["JWT_REVOKED"] = "jwt-revoked";
  GenericErrors2["INVALID_API_KEY"] = "invalid-api-key";
  GenericErrors2["INVALID_SIGNATURE"] = "invalid-signature";
  GenericErrors2["INVALID_IP"] = "invalid-ip";
  GenericErrors2["MULTIPLE_AUTH_METHODS"] = "multiple-auth-methods";
  GenericErrors2["INVALID_ID"] = "invalid-id";
  GenericErrors2["INVALID_JSON"] = "invalid-json";
  GenericErrors2["WRONG_VERB"] = "wrong-verb";
  GenericErrors2["ACCEPT_HEADER"] = "accept-header";
  GenericErrors2["CONTENT_TYPE_HEADER"] = "content-type-header";
  GenericErrors2["UNAVAILABLE_LEGAL"] = "unavailable-legal";
  GenericErrors2["APPLICATION_SUSPENDED"] = "application-suspended";
  return GenericErrors2;
})(GenericErrors || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  GenericErrors
});
