// @vonage/sms/dist/cjs/enums/SMSErrors.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/SMSErrors.ts
var SMSErrors_exports = {};
__export(SMSErrors_exports, {
  SMSErrors: () => SMSErrors
});
module.exports = __toCommonJS(SMSErrors_exports);
var SMSErrors = /* @__PURE__ */ ((SMSErrors2) => {
  SMSErrors2["SUCCESS"] = "0";
  SMSErrors2["THROTTLED"] = "1";
  SMSErrors2["MISSING_REQUIRED_PARAM"] = "2";
  SMSErrors2["INVALID_PARAM"] = "3";
  SMSErrors2["INVALID_CREDENTIALS"] = "4";
  SMSErrors2["INTERNAL_ERROR"] = "5";
  SMSErrors2["INVALID_MESSAGE"] = "6";
  SMSErrors2["BARRED_NUMBER"] = "7";
  SMSErrors2["PARTNER_ACCOUNT_BARRED"] = "8";
  SMSErrors2["PARTNER_QUOTA_VIOLATION"] = "9";
  SMSErrors2["TOO_MANY_EXISTING_BINDS"] = "10";
  SMSErrors2["ACCOUNT_NOT_ENABLED"] = "11";
  SMSErrors2["MESSAGE_TOO_LONG"] = "12";
  SMSErrors2["INVALID_SIGNATURE"] = "14";
  SMSErrors2["INVALID_SENDER_ADDRESS"] = "15";
  SMSErrors2["INVALID_NETWORK_CODE"] = "22";
  SMSErrors2["INVALID_CALLBACK_URL"] = "23";
  SMSErrors2["NON_WHITELISTED_DESTINATION"] = "29";
  SMSErrors2["SIGNATURE_AND_API_SECRET_DISALLOWED"] = "32";
  SMSErrors2["NUMBER_DEACTIVATED"] = "33";
  return SMSErrors2;
})(SMSErrors || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SMSErrors
});
