// @vonage/sms/dist/cjs/enums/SMSStatus.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/SMSStatus.ts
var SMSStatus_exports = {};
__export(SMSStatus_exports, {
  SMSStatus: () => SMSStatus
});
module.exports = __toCommonJS(SMSStatus_exports);
var SMSStatus = /* @__PURE__ */ ((SMSStatus2) => {
  SMSStatus2["SUCCESS"] = "0";
  SMSStatus2["THROTTLED"] = "1";
  SMSStatus2["MISSING_REQUIRED_PARAM"] = "2";
  SMSStatus2["INVALID_PARAM"] = "3";
  SMSStatus2["INVALID_CREDENTIALS"] = "4";
  SMSStatus2["INTERNAL_ERROR"] = "5";
  SMSStatus2["INVALID_MESSAGE"] = "6";
  SMSStatus2["BARRED_NUMBER"] = "7";
  SMSStatus2["PARTNER_ACCOUNT_BARRED"] = "8";
  SMSStatus2["PARTNER_QUOTA_VIOLATION"] = "9";
  SMSStatus2["TOO_MANY_EXISTING_BINDS"] = "10";
  SMSStatus2["ACCOUNT_NOT_ENABLED"] = "11";
  SMSStatus2["MESSAGE_TOO_LONG"] = "12";
  SMSStatus2["INVALID_SIGNATURE"] = "14";
  SMSStatus2["INVALID_SENDER_ADDRESS"] = "15";
  SMSStatus2["INVALID_NETWORK_CODE"] = "22";
  SMSStatus2["INVALID_CALLBACK_URL"] = "23";
  SMSStatus2["NON_WHITELISTED_DESTINATION"] = "29";
  SMSStatus2["SIGNATURE_AND_API_SECRET_DISALLOWED"] = "32";
  SMSStatus2["NUMBER_DEACTIVATED"] = "33";
  return SMSStatus2;
})(SMSStatus || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SMSStatus
});
