// @vonage/applications/dist/cjs/enums/VoiceRegions.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/VoiceRegions.ts
var VoiceRegions_exports = {};
__export(VoiceRegions_exports, {
  VoiceRegions: () => VoiceRegions
});
module.exports = __toCommonJS(VoiceRegions_exports);
var VoiceRegions = /* @__PURE__ */ ((VoiceRegions2) => {
  VoiceRegions2["NA_EAST"] = "na-east";
  VoiceRegions2["NA_WEST"] = "na-west";
  VoiceRegions2["EU_WEST"] = "eu-west";
  VoiceRegions2["EU_EAST"] = "eu-east";
  VoiceRegions2["APAC_SNG"] = "apac-sng";
  VoiceRegions2["APAC_AUSTRALIA"] = "apac-australia";
  return VoiceRegions2;
})(VoiceRegions || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  VoiceRegions
});
