// @vonage/number-insights/dist/cjs/enums/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/index.ts
var enums_exports = {};
__export(enums_exports, {
  CallerType: () => CallerType,
  LookupOutcome: () => LookupOutcome,
  NetworkType: () => NetworkType,
  Reachable: () => Reachable,
  ValidNumber: () => ValidNumber
});
module.exports = __toCommonJS(enums_exports);

// lib/enums/CallerType.ts
var CallerType = /* @__PURE__ */ ((CallerType2) => {
  CallerType2["UNKNOWN"] = "unknown";
  CallerType2["BUSINESS"] = "business";
  CallerType2["CONSUMER"] = "consumer";
  return CallerType2;
})(CallerType || {});

// lib/enums/LookupOutcome.ts
var LookupOutcome = /* @__PURE__ */ ((LookupOutcome2) => {
  LookupOutcome2[LookupOutcome2["SUCCESS"] = 0] = "SUCCESS";
  LookupOutcome2[LookupOutcome2["PARTIAL_SUCCESS"] = 1] = "PARTIAL_SUCCESS";
  LookupOutcome2[LookupOutcome2["FAILED"] = 2] = "FAILED";
  return LookupOutcome2;
})(LookupOutcome || {});

// lib/enums/NetworkType.ts
var NetworkType = /* @__PURE__ */ ((NetworkType2) => {
  NetworkType2["MOBILE"] = "mobile";
  NetworkType2["LANDLINE"] = "landline";
  NetworkType2["LANDLINE_PREMIUM"] = "landline_premium";
  NetworkType2["LANDLINE_TOLLFREE"] = "landline_tollfree";
  NetworkType2["VIRTUAL"] = "virtual";
  NetworkType2["UNKNOWN"] = "unknown";
  NetworkType2["PAGER"] = "pager";
  return NetworkType2;
})(NetworkType || {});

// lib/enums/Reachable.ts
var Reachable = /* @__PURE__ */ ((Reachable2) => {
  Reachable2["UNKNOWN"] = "unknown";
  Reachable2["REACHABLE"] = "reachable";
  Reachable2["UNDELIVERABLE"] = "undeliverable";
  Reachable2["ABSENT"] = "absent";
  Reachable2["BAD_NUMBER"] = "bad_number";
  Reachable2["BLACKLISTED"] = "blacklisted";
  return Reachable2;
})(Reachable || {});

// lib/enums/ValidNumber.ts
var ValidNumber = /* @__PURE__ */ ((ValidNumber2) => {
  ValidNumber2["UNKNOWN"] = "unknown";
  ValidNumber2["VALID"] = "valid";
  ValidNumber2["NOT_VALID"] = "not_valid";
  ValidNumber2["INFERRED"] = "inferred";
  ValidNumber2["INFERRED_NOT_VALID"] = "inferred_not_valid";
  return ValidNumber2;
})(ValidNumber || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CallerType,
  LookupOutcome,
  NetworkType,
  Reachable,
  ValidNumber
});
