// @vonage/number-insights/dist/cjs/enums/NetworkType.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/NetworkType.ts
var NetworkType_exports = {};
__export(NetworkType_exports, {
  NetworkType: () => NetworkType
});
module.exports = __toCommonJS(NetworkType_exports);
var NetworkType = /* @__PURE__ */ ((NetworkType2) => {
  NetworkType2["MOBILE"] = "mobile";
  NetworkType2["LANDLINE"] = "landline";
  NetworkType2["LANDLINE_PREMIUM"] = "landline_premium";
  NetworkType2["LANDLINE_TOLLFREE"] = "landline_tollfree";
  NetworkType2["VIRTUAL"] = "virtual";
  NetworkType2["UNKNOWN"] = "unknown";
  NetworkType2["PAGER"] = "pager";
  return NetworkType2;
})(NetworkType || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  NetworkType
});
