// @vonage/number-insights/dist/cjs/enums/Reachable.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/Reachable.ts
var Reachable_exports = {};
__export(Reachable_exports, {
  Reachable: () => Reachable
});
module.exports = __toCommonJS(Reachable_exports);
var Reachable = /* @__PURE__ */ ((Reachable2) => {
  Reachable2["UNKNOWN"] = "unknown";
  Reachable2["REACHABLE"] = "reachable";
  Reachable2["UNDELIVERABLE"] = "undeliverable";
  Reachable2["ABSENT"] = "absent";
  Reachable2["BAD_NUMBER"] = "bad_number";
  Reachable2["BLACKLISTED"] = "blacklisted";
  return Reachable2;
})(Reachable || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Reachable
});
