// @vonage/number-insights/dist/cjs/enums/ValidNumber.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/enums/ValidNumber.ts
var ValidNumber_exports = {};
__export(ValidNumber_exports, {
  ValidNumber: () => ValidNumber
});
module.exports = __toCommonJS(ValidNumber_exports);
var ValidNumber = /* @__PURE__ */ ((ValidNumber2) => {
  ValidNumber2["UNKNOWN"] = "unknown";
  ValidNumber2["VALID"] = "valid";
  ValidNumber2["NOT_VALID"] = "not_valid";
  ValidNumber2["INFERRED"] = "inferred";
  ValidNumber2["INFERRED_NOT_VALID"] = "inferred_not_valid";
  return ValidNumber2;
})(ValidNumber || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ValidNumber
});
